---
title: ""
date: "\\vspace{0 mm}"
csl: csl/the-plant-journal.csl
header-includes:
  \usepackage{graphicx}
  \usepackage{caption}
  \usepackage{fixltx2e}
  \usepackage{rotating}
  \usepackage{siunitx}
  \renewcommand{\thefigure}{\arabic{figure}}
  \renewcommand{\thetable}{\arabic{table}}
documentclass: tufte-handout
classoption: a4paper
output:
  rmarkdown::tufte_handout:
    dev: CairoPDF
    fig_crop: no
bibliography: bib/papersLib.bibtex
---

```{r Setup, include=FALSE, results="hide", warning=FALSE}
## grep queries:
  # bold: \*\*(\w*)\*\* -> \\\\textbf{\1}
  # italic: \*(\w*)\* -> \\\\emph{\1}
knitr::opts_chunk$set(fig.path='figures/',
               cache.path='cache/',
               fig.align = 'center', 
               dpi=600,
               fig.show='hold',
               fig.width=3.150,
               fig.height=8.661,
               echo=FALSE,
               results="hide",
               message=FALSE,
               warning=FALSE)

source('src/R/figures.R')
```

\begin{figure*}
 \includegraphics{data/lmdFigure.pdf}
\end{figure*}
\captionof{figure}{Morphology of the early stages of inflorescence development and laser microdissection of meristem samples. Toluidine blue stained sections of developing panicles at rachis meristem (RM; \textbf{a}), primary branch meristem (PBM; \textbf{b}), elongating primary branch meristem with axillary merisem (ePBM/AM; \textbf{c}) and spikelet meristem (SM; \textbf{d}) stages of differentiation. Laser microdissection (LMD) samples were collected from RM (\textbf{e} and \textbf{i}), PBM (\textbf{f} and \textbf{j}), ePBM/AM (\textbf{g} and \textbf{k}) and SM (\textbf{h} and \textbf{l}). Images show the samples before (\textbf{e}--\textbf{f}) and after (\textbf{i}--\textbf{l}) dissection. Scale bars represent \SI{100}{\micro\metre} (black, \textbf{a}--\textbf{d}), \SI{200}{\micro\metre} (red, \textbf{e}--\textbf{g} and \textbf{i}--\textbf{k}) or \SI{320}{\micro\metre} (blue, \textbf{h} and \textbf{l}).}
\clearpage

\begin{figure*}
  \includegraphics{data/isFigure.pdf}
\end{figure*}
\captionof{figure}{RNA \emph{in situ} hybridization analysis to confirm the specific expression patterns of four genes detected in the RNA sequencing dataset. Expression of \emph{LOC\_Os09g27730} (a--d), \emph{LOC\_Os01g04670} (e--h), \emph{LOC\_Os10g04270} (i--l) and \emph{LOC\_Os10g05990} (m--p) was analysed at the rachis meristem (a, e, i and m), primary branch meristem (b, f, j and n), elongating primary branch meristem plus axillary meristem (c, g, k and o) and spikelet meristem (d, h, l and p) stages. Each gene was detected in the same stage by \emph{in situ} hybridization as by RNA sequencing. The scale bars represent \SI{100}{\micro\metre}.}


```{r f_mfuzzClusters, fig.height = 8.661 / 2, fig.cap = paste0("Common patterns of gene expression in developing inflorescences. Fuzzy \\emph{c}-means clustering of normalised, variance-stabilised read counts was used to recover ", as.character(english::english(c)), " common patterns of expression. Each line describes the expression pattern of one gene, with the gene's membership to the cluster represented by the colour of the line. The core values for each cluster are plotted in black. The ordering of the panels in the plot was rearranged to enable side-by-side comparison of complementary clusters.")}
print(f_mfuzzClusters)
```

```{r f_gsea, fig.height = 8.661 / 2, fig.cap = "Geneset enrichment analysis (GSEA) for transcription factor and other transcriptional regulator families. Test statistics were calculated from the L\\textsubscript{2}FC values between the stage of interest and all other stages for the expressed genes in each geneset. The absolute value of the test statistic indicates the magnitude of the geneset-level change, and a positive test statistic indicates enrichment and a negative statistic indicates depletion. Although some test statistics had low \\emph{p}-values none were lower than 0.10 after Benjamini-Hochberg correction."}
print(f_gsea)
```

```{r f_hb, fig.height = 8.661, fig.cap = "Patterns of expression of homeobox genes that were detected in the meristem samples. The heatscale represents the geometric mean of scaled, VST-transformed read counts, and the coloured bar next to the \\emph{y}-axis shows the homeobox subfamily for each gene (Jain \\emph{et al}., 2008). The number of genes detected and the total number of genes for each subfamily is given in brackets after the subfamily name in the legend. The expression patterns were separated into five groups using hierarchical clustering. The set of homeobox genes with lowest expression in the RM and highest expression in indeterminate axillary meristems contains \\emph{OSH1} and five of the eight expressed class IV HD-Zip genes."}
print(f_hb)
```

```{r f_alogFamily, fig.width = 6.614, fig.height = 6.614, fig.cap = paste0("Expression of selected genes in the RNAseq dataset in transcripts per million (TPM). A red point indicates that the gene was above the detection cutoff, and blue indicates that the gene was not detected. (\\textbf{a}) Two reported \\emph{STMADS11}-like (\\emph{SVP}-like) targets of G1L5, \\emph{MADS22} and \\emph{MADS55} (Yoshida \\emph{et al}., 2013), are more strongly expressed in the RM than in other meristems, and the other \\emph{STMADS11}-like gene, \\emph{MADS47}, was only detected in one library. (\\textbf{b}) Only three ALOG genes (\\emph{G1L1}, \\emph{G1L2} and \\emph{G1L5}/\\emph{TAWAWA1}) were detected, and they share a similar pattern of expression. (\\textbf{c}) Several \\emph{SPL} genes and a co-regulated microRNA precursor, \\emph{MIR319A}, which was recovered in cluster ", getClusterName('small', 'dec')," are highly expressed in RM and PBM before decreasing in expression in ePBM/AM and SM. (\\textbf{d}) \\emph{GN1A} (\\emph{CKX2}) was not detected in the LMD dataset, but \\emph{LOG1} is expressed in all meristem types, and two other genes possibly related to cytokinin activation, \\emph{LABA1} (\\emph{LOGL6}) and \\emph{LOGL1}, are expressed more highly in ePBM/AM and SM than in RM and PBM.")}
# set layout
grid::grid.newpage()
grid::pushViewport(grid::viewport(layout = grid::grid.layout(4, 4)))

# print the plots
print(f_alogFamily_a, vp = grid::viewport(layout.pos.row = 1:3, layout.pos.col = 1))
print(f_alogFamily_b, vp = grid::viewport(layout.pos.row = 1:3, layout.pos.col = 2))
print(f_alogFamily_c, vp = grid::viewport(layout.pos.row = 1:3, layout.pos.col = 3:4))
print(f_alogFamily_d, vp = grid::viewport(layout.pos.row = 4, layout.pos.col = 1:4))
```


