---
title: ""
csl: csl/the-plant-journal.csl
header-includes:
  \usepackage{graphicx}
  \usepackage[utf8]{inputenc}
  \DeclareUnicodeCharacter{2265}{$\geq$}
output:
  word_document:
    fig_caption: yes
    reference_docx: csl/wordStyles.docx
bibliography: bib/papersLib.bibtex
---


```{r Setup, include=FALSE, results="hide", warning=FALSE}
knitr::opts_chunk$set(fig.path='figures/',
               cache.path='cache/',
               fig.align = 'center', 
               dpi=600,
               fig.show='hold',
               fig.height = 4.213,
               fig.width = 4.213,
               echo=FALSE,
               results="hide",
               message=FALSE,
               warning=FALSE)

source('src/R/figures.R')
```

# `r I(pasteLabel("Figure S", s_figCount, "sf_pca"))`

Principal components analysis of transformed read counts for each library. The two libraries made from degraded RNA did not cluster with other libraries (see `r I(pasteLabel("Table S", s_tableCount, "st_libStats"))`), suggesting that the use of degraded RNA affected quantification of expression. For this reason these two libraries were not included in subsequent analyses.

# `r I(pasteLabel("Figure S", s_figCount, "sf_isGenesTpm"))`

Comparison of the LMD RNA-sequencing dataset with published *in situ* results. Expression value in transcripts per million (TPM) is shown on the *y*-axis, and points are plotted as a triangle if the expression value was above the cutoff for that library or a circle if below. Grey shading indicates a meristem type for which expression was detected by *in situ* hybridisation. See `r I(pasteLabel("Table S", s_tableCount, "st_refTable"))` for references for the *in situ* expression patterns. Although four replicates were produced from each of the RM and PBM, only three replicates for each stage are included in the analysis (see Results).

Gene expression measured by RNAseq may differ from the qualitative patterns measured by RNA *in situ* hybridization for several reasons, including the relative sensitivity of the techniques, the specificity of the *in situ* probe, and differences in growth conditions (*e.g. HD5*) or the accession under study (*e.g. GN1A*).

# `r I(pasteLabel("Figure S", s_figCount, "sf_mfuzzPcaCentroids"))`

Assessment of clustering results. **a**. Minimum centroid distance for 2â€“`r maxClust` cluster cores (*c*). Eight cluster cores were indicated because the rate of change of the minimum centroid distance over *c* decreased after *c* = 8. **b**. Classical multidimensional scaling (MDS) of Euclidean distances between transformed read counts for clustered genes. Genes are coloured by the cluster to which they were assigned. The separation of genes by cluster on the MDS plot supports the use of eight clusters.

# `r I(pasteLabel("Figure S", s_figCount, "sf_madsTree"))`

Differential expression of genes encoding putative MIKC proteins in *A. thaliana*, *S. lycopersicum* and *O. sativa* ssp. *japonica*. The tree is a neighbour-joining tree estimated from genetic distance of amino acid sequences (see Methods S1). Named proteins starting with AT_ or OS_ are from Arabidopsis or rice respectively, with synonyms in parentheses, and other proteins are labelled with their unique identifiers.

The colour of the label represents the Log~2~-fold change (L~2~FC) in expression of the gene encoding that protein between undifferentiated and differentiated meristems. Where there is no coloured label, the adjusted *p*-value associated with the difference in expression was greater than 0.1, or the gene for the protein in question was excluded from the differential expression analysis by `DESeq2`, which may occur because of low read counts, an extreme count outlier or the gene failing independent filtering [@Love:2014do].

For rice, L~2~FC was estimated from the RM and SM samples reported here. For *A. thaliana*, L~2~FC was estimated using the inflorescence meristem and floret meristem samples reported by @Mantegazza:2014di. The sympodial inflorescence meristem and floret meristem samples generated by @Park:2012dg were used to estimate L~2~FC values for tomato genes.

# `r I(pasteLabel("Table S", s_tableCount, "st_libStats"))`

Library and sequencing statistics.

# `r I(pasteLabel("Table S", s_tableCount, "st_refTable"))`

References for `r I(pasteLabel("Figure S", s_figCount, "sf_isGenesTpm"))`.

```{r insituRefs, results = "asis"}
pander::pandoc.table(st_refTable, emphasize.cols = c(1,2))
```

# Data S1

Expression values for all genes in transcripts per million with annotations.

# Data S2

List of clustered genes with annotations.

# References for Supporting Information