---
title: ""
csl: csl/the-plant-journal.csl
header-includes:
  \usepackage{graphicx}
  \usepackage[utf8]{inputenc}
  \DeclareUnicodeCharacter{2265}{$\geq$}
output:
  word_document:
    fig_caption: yes
    reference_docx: csl/wordStyles.docx
bibliography: bib/papersLib.bibtex
---

```{r Setup, include=FALSE, results="hide", warning=FALSE}
knitr::opts_chunk$set(fig.path='figures/',
               cache.path='cache/',
               fig.align = 'center', 
               dpi=600,
               fig.show='hold',
               fig.height = 4.213,
               fig.width = 4.213,
               echo=FALSE,
               results="hide",
               message=FALSE,
               warning=FALSE)

source('src/R/figures.R')
```

# Methods S1: Comparative analysis of the MADS family

The primary sequences of *A. thaliana*, *S. lycopersicum* and *O. sativa* MADS proteins were downloaded by querying Phytozome 10.3 [@Goodstein:2012ho] for the PANTHER family PTHR11945 ('MADS BOX PROTEIN') via the `biomaRt` interface [@Thomas:2003db; @Durinck:2009ki]. An initial alignment of all proteins was performed with `Clustal Omega` using a full distance matrix [@Sievers:2011fn]. To clean the alignment, poorly conserved regions where greater than `r minpcnongap` percent of sequences had gaps or less than `r minpcident` percent of pairwise comparisons between sequences had identical residues were removed. To recover MADS genes related to the Arabidopsis MIKC clade, a UGPMA tree was estimated from the cleaned alignment and the tree was cut at the lowest height that returned a clade including all type II Arabidopsis MADS proteins listed in the Arabidopsis transcription factor database on the Arabidopsis Gene Regulatory Information Server [@Yilmaz:2011ew], with as few type I proteins as possible. Protein sequences in this clade that were longer than `r minProtLength` amino acids were realigned using `Clustal Omega`, and the alignment was cleaned as described above and used to estimate a neighbour-joining tree with the `bionj` algorithm implemented in the `ape` package [@Paradis:2004dv] rooted to type I MADS sequences (`r og`). L~2~FC change expression values for tomato and Arabidopsis genes were estimated from samples that have previously been described [@Mantegazza:2014di; @Park:2012dg] using the same `STAR`â€“`DESeq2` pipeline described in the main text for the rice samples [@Dobin:2012fg; @Love:2014do]. The tree was plotted using the `ggtree` extension for `ggplot2` [https://github.com/GuangchuangYu/ggtree; @Wickham:2009tn], with clade labels following the nomenclature presented by @Arora:2007fa.

# References for Methods S1 