---
title: Changes in gene expression associated with transitions in reproductive meristem identity during early development of rice inflorescence.
csl: csl/the-plant-journal.csl
header-includes:
  \usepackage{graphicx}
  \usepackage[utf8]{inputenc}
  \DeclareUnicodeCharacter{2265}{$\geq$}
output:
  BiocStyle::html_document:
    dev: CairoPDF
    fig_caption: yes
    force_captions: yes
    highlight: pygments
    number_sections: yes
  word_document:
    fig_caption: yes
    reference_docx: csl/wordStyles.docx
    toc: yes
bibliography: bib/papersLib.bibtex
---

```{r Setup, include=FALSE, results="hide", warning=FALSE}

# max width (2 col): 6.614
# max width (1 col): 3.150
# max height : 8.661

knitr::opts_chunk$set(fig.path='figures/',
               cache.path='cache/',
               dev=c("png", "cairo_pdf"),
               fig.width=3.150,
               fig.height=8.661,
               fig.align = 'center',
               dpi=300,
               fig.show='hold',
               cache=FALSE,
               echo=FALSE,
               results="hide",
               message=FALSE,
               warning=FALSE)

pander::panderOptions("table.caption.prefix", "")

source('src/R/figures.R')

```

Thomas W.R. Harrop^§,1^, Israr Ud Din^¶,^[^1], Veronica Gregis^¶^, Michela Osnato^¶^, Stefan Jouannic^§^, Hélène Adam^§,2^ and Martin Kater^¶,^[^2]

[^1]: These authors contributed equally to this work.
[^2]: To whom correspondence should be addressed: martin.kater@unimi.it or helene.adam@ird.fr.

| ^§^ Institut de Recherche pour le Développement, 911 Avenue Agropolis, 34394 Montpellier, France.
| ^¶^ Università degli Studi di Milano, Via Celoria 26, 20133 Milano, Italy.

## Running title

## Keywords

## SRA details for libraries

## Word count

# Summary

In rice, inflorescence architecture is established at early stages of reproductive development by the activity of different meristem types, and contributes directly to grain yield potential. After induction of flowering, the shoot apical meristem is converted to an inflorescence meristem to form the rachis, which is the main axis of the inflorescence. The rachis meristem produces primary branch meristems, which themselves produce axillary meristems during elongation. The axillary meristems may become secondary or higher-order branch meristems, retaining their indeterminate state, or be converted directly to determinate spikelet meristems. After elongation, the primary and higher-order branch meristems differentiate into spikelet meristems, and the complexity of branching and number of spikelets is fixed. Although some of the genes involved in transitions in meristem identity in rice have been identified, understanding of the network of genes controlling this process is lacking. To address this, we used a precise laser microdissection and RNA sequencing approach in *Oryza sativa* ssp. *japonica* cv. Nipponbare to produce quantitative data that describe the landscape of gene expression in four different meristem types. A switch in expression profile between apical and axillary meristem types followed by more gradual changes during transitions in axillary meristem identity was observed and a number of potential candidates were identified. This resource will be vital for a mechanistic understanding of the link between inflorescence development and grain yield that integrates gene expression with environmental and other epigenetic factors, as well as for comparative studies of the domestication process.

# Significance statement

# Introduction

Agricultural development is essential to ensure food production and security for a growing population [@Borlaug:2007cp]. Rice is a staple food for over half the world's population, including many developing countries, and a sustainable increase in the production of rice under the constraints of a changing climate and diminishing water and land availability will require plants with improved grain output, making the establishment of high-yield rice varieties a goal of modern breeding programs [@Peng:2008jb]. Rice yield is a complex trait influenced by genetic and epigenetic factors, and is progressively defined during the life cycle of the plant, first during the vegetative phase, where the number of fertile tillers is established, and then during the reproductive phase and grain-filling phase. Yield is directly dependent on the architecture of the inflorescence [@Ikeda:2004tk].

The branched inflorescence of rice is a compound raceme, classified as a panicle, composed of a rachis (the main axis), primary branches, higher order branches, and spikelets. During panicle development, the undifferentiated cells in the primary axis (*i.e.* the rachis meristem, RM) undergo transitions through a series of meristem identities. After the establishment of the rachis some RM cells differentiate into primary branch meristem (PBM) for the formation and elongation of primary branches. During elongation, the PBM is indeterminate and can give rise to axillary meristems, which may differentiate into spikelets or secondary and higher-order branches. The PBM and secondary branch meristem (SBM) are both converted to a terminal spikelet meristem (SM). Terminal and axillary SMs develop into floret meristems (FM) and produce a single floret, meaning that grain yield is determined by the number of spikelets [@Ikeda:2004tk; @Yoshida:2011bo]. After spikelet differentiation, panicle branching complexity is fixed and the rachis and branches elongate rapidly and heading and flowering occurs [@Ikeda:2004tk].

The landscape of gene expression is a characteristic factor that differentiates meristem cell types, and is involved in the control of transitions between meristem identities and therefore panicle phenotype. Mutant analyses and map-based cloning of quantitative trait loci have identified a number of genes required for the initiation and development of panicles, as well as genes controlling the number and size of grains and panicles [@Xing:2010de; @Wang:2011kq]. Some of these genes are involved in the patterning of axillary meristem and panicle branching, such as the nuclear regulatory factor-encoding genes *MONOCULM 1*, *LAX PANICLE 1* and *LAX PANICLE 2* [@Wang:2011kq]. A large set of genes related to floral development, which may also affect panicle architecture, has been identified [@Yoshida:2011bo], some of which may have been under selection during domestication or are associated with crop improvement in Asian rice  [@Xing:2010de; @He:2011io; @Wang:2011kq; @Xu:2012hq; @Ikeda:2013go].

Understanding the events that occur during development that determine panicle characteristics, such as branching complexity and its plasticity, will be vital for sustainable improvement of rice yield potential using targeted breeding programs. Despite advances in characterization at the level of individual genes and interactions, a complete understanding of the control of panicle morphology and grain yield will require mechanistic studies explaining the interactions of gene regulatory network components with each other and the environment [@Azpeitia:2013jia]. One step towards this goal is to describe the changes in gene expression between different cell types during development. In this article, we describe the measurement and analysis of genome-wide expression in meristematic tissues from the early stages of rice panicle development, using a precise laser microdissection and RNA sequencing approach.

# Results

## Genome-wide expression analysis of inflorescence development in rice

To investigate gene expression during the development of the rice inflorescence, laser microdissection (LMD) was used to collect meristematic tissues from the rachis meristem (RM), primary branch meristem (PBM), elongating primary branch meristem with secondary branch meristem (ePBM/SBM) and spikelet meristem (SM) of early panicles (`r I(pasteLabel("Figure ", figCount, "f_lmdFigure"))`). RNA isolated from the meristem samples was amplified and used to produce cDNA libraries for sequencing. RNA input for amplification was between `r min(st_libStats$"Yield (ng)")` and `r max(st_libStats$"Yield (ng)")` ng and the average RNA integrity number (RIN) was `r round(mean(st_libStats$"RIN"), 1)` (`r I(pasteLabel("Table S", s_tableCount, "st_libStats"))`; **Number of panicles used for each library**). Initially, two biological replicates were prepared for each stage, but after performing a principal components analysis (PCA) on transformed expression values, two libraries that were produced from degraded RNA (RIN < 6.5) were excluded from the analysis (`r I(pasteLabel("Figure S", s_figCount, "sf_pca"))`). Following this, further samples were prepared to provide a total of three biological replicates with intact RNA (RIN ≥ 7) for each meristem type.

At least `r round(min(st_libStats$"Reads (M)"), 0)` million single-end, 50-base reads were produced from each library, yielding between `r round(min(st_libStats$"Reads in genes (M)", na.rm = TRUE), 0)` and `r round(max(st_libStats$"Reads in genes (M)", na.rm = TRUE), 0)` million reads that mapped to genes after excluding reads likely to have originated from rRNA and tRNA. Using strict cut-offs for gene expression, `r length(unique(readRDS("output/expressedGenes/expressedGenesAll.Rds")))` unique genes were detected in two or more libraries from at least one stage. As a preliminary assessment of the RNA sequencing (RNAseq) results, the qualitative and quantitative expression of a selection of genes was compared with patterns that had previously been reported. Although there were several exceptions, most genes that were detected by *in situ* hybridisation were also detected in the LMD dataset (`r I(pasteLabel("Figure S", s_figCount, "sf_isGenesTpm"))` and `r I(pasteLabel("Table S", s_tableCount, "st_refTable"))`).  Four genes that were detected in specific meristem samples were selected for validation by *in situ* hybridization (`r I(pasteLabel("Figure S", s_figCount, "sf_inSitu"))`). **How do in situ results compare to transcriptome?**

##  Patterns of gene expression during inflorescence development

To recover common expression patterns in expressed genes, read counts were transformed using the variance-stabilizing transformation (VST) included in the `DESeq2` software package [@Love:2014do]. Detected genes were ranked by variance, and standardized, VST-transformed read counts for the `r as.character(dim(expressionMatrix)[1])` most-variable genes (33% of expressed genes) were clustered using the fuzzy *c*-means algorithm implemented in the `Mfuzz` package [@Kumar:2007uw]. To determine the number of cluster cores (*c*), the minimum distance between cluster centroids, the formation of empty clusters, and PCA plots of cluster members were monitored for clusters produced with *c* values between 2 and `r maxClust` (`r I(pasteLabel("Figure S", s_figCount, "sf_mfuzzPcaCentroids"))`), leading to the use of `r as.character(english::english(c))` cluster cores (`r I(pasteLabel("Figure ", figCount, "f_mfuzzClusters"))`). Several common patterns were recovered by clustering: genes that increase or decrease in expression steadily during development (clusters `r getClusterName('inc', 'inc')` and `r getClusterName('dec', 'dec')`), genes that change expression between the apical meristem (RM) and axillary meristem (PBM–SM) samples (clusters `r getClusterName('inc', 'small')` and `r getClusterName('dec', 'small')`) and genes that change expression gradually over the course of changes in axillary meristem identity (clusters `r getClusterName('small', 'inc')` and `r getClusterName('small', 'dec')`). There were also two weaker clusters containing fewer genes, which had complex expression patterns involving changes in expression particular to the PBM (clusters `r getClusterName('inc', 'dec')` and `r getClusterName('dec', 'inc')`).

Examination of the clustered genes suggested that transcription factor (TF) genes were overrepresented in most clusters (Data S1). To determine whether particular TF families were associated with particular patterns of expression, the enrichment of TF family members in each cluster was calculated using hypergeometric tests. Using the list of transcription factor and other transcriptional regulator families available on the Plant Transcription Factor Database [@PerezRodriguez:2010co], a hypergeometric test was carried out if there were at least two members of one family in a given cluster. After adjustment for multiple comparisons, most clusters had a significant enrichment of at least one family (`r I(pasteLabel("Table ", tableCount, "t_hypergeom"))`). For example, cluster `r getClusterName('inc', 'small')`, containing genes that are more highly expressed in the axillary meristem (PBM–SM) samples than in the RM sample, has an enrichment of Homeobox and basic helix-loop-helix TF genes, whereas the cluster with the complementary pattern (cluster `r getClusterName('dec', 'small')`) is enriched for ABI3VP1 TFs and Aux/IAA auxin response genes. The enrichment of different TF families in different clusters suggests complex control of the landscape of gene expression during inflorescence development.

## Dynamic expression of transcription factor families

Because of the enrichment of TF family members in some of the gene expression clusters, geneset enrichment analysis (GSEA) was used to provide an overview of expression of transcription factors and other regulators in the meristem samples (`r I(pasteLabel("Figure ", figCount, "f_gsea"))`). Unlike soft clustering, which was used to group genes based on shared expression dynamics, this analysis was used to investigate the overall strength of expression of the families based on Log~2~-fold changes (L~2~FC) in read count of their members. Some  modes of expression shared between different families may suggest roles in inflorescence development. The most pronounced change in expression appears to occur between the RM and branch meristem samples (*i.e.* the change from apical to axillary meristem). Several families (including ABI3VP1 and GROWTH-REGULATING FACTOR TF genes and Aux/IAA genes) are more highly expressed in RM than in the other meristem types. Another group, including MYB and SBP TF genes and SET and PHD regulator genes, is enriched in one or both of the PBM and ePBM/SBM samples but depleted or not enriched in the RM samples.

# Discussion

## Switch in gene expression between apical and axillary meristems

Two results are presented here that suggest significant changes in gene expression between apical and axillary meristems: a difference in expression of annotated transcription factor genes between RM and PBM samples, and the recovery of co-expression clusters of primarily uncharacterized genes that appear to be either switched on or off between the RM and the PBM. Clusters `r getClusterName('dec', 'small')` and `r getClusterName('inc', 'small')` contain genes that change in expression between apical (RM) and axillary meristems (PBM–SM). Axillary meristem initiation requires auxin synthesis and transport in *A. thaliana* and maize [reviewed by @Gallavotti:2013ki], but these processes are not as well understood in rice. Cluster `r getClusterName('inc', 'small')` contains the auxin response factor *ARF6A* and the small auxin-up RNA *SAUR33*, as well as two ABC transporters, while the complementary cluster `r getClusterName('dec', 'small')` contains four uncharacterized genes with annotations relating to auxin (*IAA10* and *IAA16*, two auxin-responsive transcriptional regulators; *ABCB27*, an ABC transporter; and *LOC_Os01g63770*, an AUX/LAX transporter), and the GH3 gene related to auxin homeostasis, *GH3-8*. In *A. thaliana*, excess auxin may be inactivated by conjugation to amino acids by GH3 enzymes [@Staswick:2005cv], and transgenic rice plants containing 3 kb of upstream sequence from *FLO-LFY HOMOLOG OF RICE* (*RFL*) fused to the *GH3-8* coding sequence, presumably expressing *GH3-8* in the endogenous pattern of *RFL* [*i.e.* PBM and SBM; @IkedaKawakatsu:2012co], have smaller panicles with fewer branches [@Yadav:2011db]. GH3-8 enzyme conjugates amino acids to auxin *in vitro* [@Ding:2008hv], suggesting that the panicle phenotype may be related to reduced auxin signalling. Although auxin response factors and Aux/IAA response genes are generally more strongly expressed at earlier stages (`r I(pasteLabel("Figure ", figCount, "f_gsea"))`), the presence of auxin-related genes in complementary clusters suggests that the transcriptional effects of auxin signalling are dependent on meristem type, and understanding of the relationship between auxin and meristem specification will require functional analysis of the components of the regulatory network involved in the response in these tissues.

Overexpression of the *G1*-Like (G1L/ALOG) transcriptional activator *G1L5* (*TAWAWA1*) increases the production of secondary branches and leads to the formation of tertiary branches, resulting in higher grain yield [@Yoshida:2013ff]. G1L5 activates transcription of two *STMADS11*-like (*SVP*-like) MADS genes, *MADS22* and *MADS55*, but activation of the third rice *STMADS11*-like gene, *MADS47*, was not reported [@Yoshida:2013ff]. *MADS22* and *MADS55* also decrease in expression between PBM and SM, and *MADS55* is in the same cluster as *G1L5*, but *MADS47* is not highly expressed (`r I(pasteLabel("Figure ", figCount, "f_alogFamily"))`a). The presence and co-expression of genes that are involved in inflorescence development along with other uncharacterized TF genes indicates that the clusters capture biologically meaningful groups of genes. Of the ALOG genes, which are annotated with the Pfam domain PF04852, only three were detected in the dataset, with two of them present in clusters `r getClusterName('dec', 'small')` and `r getClusterName('dec', 'dec')` (*G1L2* and *G1L5* respectively), and the third, *G1L1*, not present in the clustering results but having a similar expression pattern (`r I(pasteLabel("Figure ", figCount, "f_alogFamily"))`b). *g1l5* missense mutants have small inflorescences with reduced grain yield, but this phenotype was stronger when *G1L5* was targeted by RNA interference [@Yoshida:2013ff]. This may suggest functional redundancy between the three co-expressed ALOG genes, *G1L1*, *G1L2* and *G1L5*, with more than one being affected by the RNAi construct used against *G1L5*. Although the increase in secondary branches linked to *G1L5* suggests that it acts by suppression of SM identity, perhaps via downstream factors that may promote BM activity such as the *STMADS11*-like genes, *g1l5* plants also have fewer primary branches [@Yoshida:2013ff], and the peak expression in the RM of *G1L1*, *G1L2*, *G1L5*, *MADS22* and *MADS55* suggests an additional role before the transition from apical to axillary meristem that remains to be reported.

## Gradual changes in expression during transitions in axillary meristem identity

The patterns of differential expression of TF genes presented in `r I(pasteLabel("Figure ", figCount, "f_gsea"))` are not intended to represent each member of each gene family. For example, the test statistic for the whole MADS family was higher in the SM than in the RM, but there are several MADS genes that have lower expression in the SM than in the RM, such as *PANICLE PHYTOMER2* (*PAP2*/*MADS34*), which was strongly expressed at all stages with higher expression in the RM and PBM. *PAP2* acts with another *AGAMOUS-LIKE2*-like (*AGL2*-like/*SEPELLATA*) MADS gene, *LEAFY HULL STERILE 1* (*LHS1*/*MADS1*), to specify floral organs [@Gao:2010iz], but *pap2* mutants also have altered primary and secondary branch number, suggesting an earlier role in spikelet meristem specification [@Gao:2010iz; @Kobayashi:2010hn]. In contrast, *LHS1*, which is involved in floral meristem determination and floral organ development [@Jeon:2000tj; @Agrawal:2005ho], was detected only in the SM samples and is thus representative of the overall enrichment of MADS genes in the SM.

Compared to the switch from apical to axillary meristem, more gradual changes in gene expression were apparent during the sequential transition from PBM to SM. The enrichment of MADS genes in the SM sample and their overrepresentation in cluster `r getClusterName('small', 'inc')` corresponds with their involvement in flower development in rice and other plants. In the eudicot *A. thaliana*, *APETALA1* (*AP1*) is involved in determining FM identity. *AP1* also interacts with members of the *AGL2*-like clade (*SEPALLATA* MADS-domain factors), *APETALA3* (*AP3*) and *PISTILLATA* (*PI*) to specify petal identity, and *AGL2*-like clade members also interact with *AGAMOUS* (*AG*) to specify stamens [recently reviewed by @Prunet:2013ki]. In rice, functions similar to those provided by *A. thaliana* *PI* and *AP3* genes, related to specification of stamens and lodicules, appear to be performed redundantly by *MADS2*, *MADS4* and *SUPERWOMAN 1* (*SPW1*/*MADS16*) [@Nagasawa:2003gu; @Yao:2008dy], whilst stamen and carpel identity are specified by the expression of *RICE AGAMOUS* (*RAG*/*MADS3*) and *MADS58* [@Kyozuka:2002vm; @Yamaguchi:2006br; **ref Dreni 2011**]. Two recent studies in *A. thaliana* and the domesticated tomato, *Solanum lycopersicum*, used LMD and RNA sequencing to study inflorescence development [@Park:2012dg; @Mantegazza:2014di]. Comparison of these datasets with the rice LMD dataset presented here suggests some differences in expression of MIKC^C^ genes between these species (`r I(pasteLabel("Figure S", s_figCount, "sf_madsTree"))`). *A. thaliana* *AGL6* is the only gene from the *AGL6*-like clade that is more highly expressed in indeterminate meristems than determinate meristems. *agl6* mutants have fewer axillary buds on the inflorescence stem, suggesting that *AGL6* promotes axillary meristem formation [@Huang:2012de]. The rice genes from the *AGL6*-like clade, *MFO* (*MADS6*) and *MADS17*, redundantly promote floral meristem determinancy [@Ohmori:2009kj], and are correspondingly enriched in determinate meristems. Expression of the putative tomato *AGL6*-like gene *SOLYC01G093960.2* is also enriched in determinate meristems, suggesting that the role of *AGL6* in axillary meristem formation may be specific to the Arabidopsis lineage. In contrast, consistent with their functions in floral organ specification in rice and *A. thaliana*, genes in the *AGL2*-like clade are generally more highly expressed in determinate meristems.

The strong enrichment of SQUAMOSA promoter binding protein-like (SPL/SBP) genes in the PBM and their presence in cluster `r getClusterName('small', 'dec')` suggests involvement in the early stages of inflorescence development. *WEALTHY FARMER'S PANICLE* (*WFP*/*SPL14* or *IDEAL PLANT ARCHITECTURE 1*/*IPA1*), *SPL3*, *SPL12* and *SPL17* are all highly expressed in the RM and PBM, gradually decrease in expression and have weakest expression in the SM (`r I(pasteLabel("Figure ", figCount, "f_alogFamily"))`c), suggesting a redundant function in indeterminate axillary meristems. These genes are all putative targets of *miR156*, a microRNA that causes reduced panicle size and delayed flowering when overexpressed [@Xie:2006dc; @Wang:2015jx], *SPL* gene expression is controlled by *miR156* and *miR529* [**ref Jeong 2011**; @Wang:2015jx]. The precursor for another microRNA, *MIR319A*, was recovered from the same cluster as *SPL3* and *SPL12* and has a similar expression pattern to these genes, *SPL17* and *WFP* (`r I(pasteLabel("Figure ", figCount, "f_alogFamily"))`c). Overexpression of *WFP*, which was detected by *in situ* hybridisation experiments in the RM and branch meristems in wild-type plants, is associated with increased branching and grain yield [@Jiao:2010ft; @Miura:2010it]. WFP binds to the promoter region and appears to cause upregulation of the *DENSE AND ERECT PANICLE 1* (*DEP1*) gene [@Lu:2013hy], which is present in cluster `r getClusterName('inc', 'dec')` and encodes a major QTL for grain production and panicle morphology in high-yield *O. sativa* ssp. *japonica* varieties. Probable loss-of-function *dep1* mutants have larger SAMs with higher cell number, and reduced expression of *GN1A* [@Huang:2009kf]. WFP also binds to the promoters of *LONELY GUY 1* (*LOG1*), *PIN PROTEIN 1B* and *SLENDER RICE1*, suggesting a role in the coordination of hormone signalling [@Lu:2013hy].

A homolog of *LOG1*, *LONELY GUY LIKE PHOSPHORIBOHYDROLASE 1* (*LOGL1*), is present in cluster `r getClusterName('inc', 'inc')` with genes that increase in expression between RM and SM. In rice, cytokinins may increase branching complexity by promoting IM or BM activity [reviewed by @Han:2014dt]. The LOG1 enzyme converts cytokinin nucleotides to the active form *in vitro*, and *log1* mutants have lower expression of two cytokinin-inducible *RESPONSE REGULATOR* genes, resulting in early termination of BM and IM and small panicles with small SAMs and branching defects [@Kurakawa:2007go]. Similarly, reduced expression of a cytokinin oxidase gene, *GRAIN NUMBER 1A* (*GN1A*/*CKX2*), which encodes an enzyme that inactivates cytokinin, is associated with increased grain production in several high-yield *O. sativa* ssp. *indica* varieties [@Ashikari:2005eg]. Although *GN1A* transcripts were not detected in this LMD dataset (`r I(pasteLabel("Figure ", figCount, "f_alogFamily"))`d), these results suggest that an increase in active cytokinin is associated with grain yield via an effect on meristem activity, but the *LONG AND BARBED AWN1* (*LABA1*/*LOGL6*/*An-2*) gene provides evidence for an inverse role. The LABA1 protein present in *O. rufipogon*, which shares an identical primary sequence with the protein from *O. sativa* ssp. *japonica* cv. Nipponbare, also activates cytokinin *in vitro* [@Gu:2015hy; @Hua:2015bs], and introduction of the *Laba1* allele **I think this is correct, because the domesticated *laba1* gene product is non-functional. Can someone pls confirm? Tom.** from *O. rufipogon* into an *O. sativa* ssp *indica* background that contains a non-functional *laba1* allele results in a higher concentration of endogenous cytokinins and higher expression of *RESPONSE REGULATOR* genes [@Hua:2015bs], but reduces the number of grains per panicle and the number of tillers per plant [@Gu:2015hy]. *laba1* is a domestication allele found at high frequencies in cultivated accessions, implying that it was affected by artificial selection either for awn phenotype or grain yield [@Gu:2015hy; @Hua:2015bs]. *LABA1* was detected in the ePBM/SBM and SM samples but not the RM or PBM samples. Although the function of LOGL1 has not been reported, the expression patterns of *LABA1* and *LOGL1* suggest increasing cytokinin activation along the course of axillary meristem determination (`r I(pasteLabel("Figure ", figCount, "f_alogFamily"))`a). Such a mechanism would support a cytokinin function secondary to the maintenance of IM and BM, but it is currently not clear how this would influence meristem identity or panicle branching at the molecular level. A type-A *RESPONSE REGULATOR* (*RR3*) is present in cluster `r getClusterName('dec', 'small')` (containing genes more highly expressed in RM), highlighting the possible complexity of the roles of cytokinins in inflorescence development.

## Homeodomain genes and meristem identity

*log1* mutants also appear to have reduced *HOMEOBOX 1* (*OSH1*) expression [@Kurakawa:2007go]. *OSH1* encodes a class I *Knotted1-like homeobox* (*KNOX*) homeodomain transcription factor. The expression of class I *KNOX* genes in young inflorescences has been detected by *in situ* hybridisation [@Sentoku:1999ti]. *osh1* mutants have defects in SAM maintenance during vegetative growth and lower expression of other *KNOX* genes, and expression of *KNOX* genes can be induced by cytokinin treatment during shoot regeneration [@Tsuda:2011gz]. Another class I *KNOX* gene, *HOMEOBOX 15* (*OSH15*) is expressed in a similar pattern to *OSH1* (`r I(pasteLabel("Figure ", figCount, "f_hb"))`), as previously reported in *O. sativa* cv. Nipponbare [@Sato:1998tp]. OSH1 binds to the upstream region of the *OSH15* locus, and *osh1 osh15* double-mutants have lower induction of *OSH6* after treatment with cytokinin [@Tsuda:2011gz], supporting a role of *KNOX* genes in the response to phytohormone, and the higher expression of *OSH1* and *OSH15* in indeterminate axillary meristems is compatible with a function in the promotion of BM identity. In *A. thaliana*, cytokinin increases the production of the homeodomain gene *WUSCHEL*, which itself enhances cytokinin signalling [@Leibfried:2005hw; @Lindsay:2006bh; @Gordon:2009fp]. The rice ortholog of *WUSCHEL*, *WUSCHEL-LIKE HOMEOBOX 1* (*WOX1*/*WUS*) [@Nardmann:2006fd], was not detected in the meristem tissues described here, but another *WUSCHEL homeobox* (*WOX*) gene, *WOX8*, has a peak in expression in RM, and a third, *LOC_Os01g70810*, is predominantly expressed in the RM and PBM (`r I(pasteLabel("Figure ", figCount, "f_hb"))`). Although these results and previous studies support a role of *KNOX* and *WOX* genes in the maintenance of indeterminate meristem identity, possibly coordinated by phytohormone signalling, more work is required to elucidate this mechanism.

Three class IV HD-Zip genes, *RICE OUTMOST CELL-SPECIFIC GENE 1* (*ROC1*), *ROC3* and *LOC_Os09g35760*, were recovered in cluster `r getClusterName('inc', 'small')`, containing genes that had lowest expression in the RM and higher expression in the samples from axillary meristems. A prominent group of homeobox genes follow a similar pattern, including five of the eight detected class IV HD-Zip genes and several genes from other homeobox subclasses, including *OSH1* (`r I(pasteLabel("Figure ", figCount, "f_hb"))`). *Zea mays* class IV HD-Zip genes are expressed in the outer cell layer of the SAM and overexpression of *OUTER CELL LAYER 1* delays flowering in transgenic plants [@Javelle:2011dz; @DepegeFargeix:2011ds]. In rice, several class IV HD-Zip genes have been detected in the vegetative SAM by *in situ* hybridisation [@Ito:2003ta], but their role in axillary meristems in the developing inflorescence has not been explained. There are also smaller groups of homeobox genes that peak in expression in the SM or RM samples, suggesting multiple roles and possible redundancy for homeodomain proteins during inflorescence development.

## Summary

This report describes the changes in gene expression that occur during the early stages of rice panicle development. A switch in expression between apical and axillary meristems followed by gradual changes between PBM and SM were observed, based on changes in expression of uncharacterized genes, transcription factors and genes with functions potentially related to hormone signalling. These results are expected to lead to mechanistic studies of the regulatory networks that establish gene expression in these tissues as well as functional studies of the individual components, both of which will be vital for understanding the genetic factors influencing rice yield via panicle morphology.

# Experimental procedures

## Plant material and sampling

*Oryza sativa* ssp. *japonica* cv. Nipponbare plants were grown in a growth chamber with 70% relative humidity at 30°C during the day and 26°C at night. After 7 weeks in long day conditions (14 hours light and 10 hours darkness) plants were transferred to short day conditions (10 hours light and 14 hours darkness) to induce floral transition. Inflorescence meristems were harvested from 6 to 14 days after the change of photoperiod.

## Tissue embedding, laser microdissection and sequencing

Each inflorescence was harvested in 2 mL ice-cold 3:1 ethanol:acetic acid fixative solution, infiltrated (**filtered?**) twice under a mild vacuum for 15 minutes and stored for 20 hours in fresh fixative at 4°C. Embedding for dissection was performed as described by @Mantegazza:2014di. Tissues were cut into 8 µm sections on an RM2155 or RM2255 microtome (Leica) and dissected on a LMD6000 or LMD7000 laser dissector (Leica). RNA was amplified using the ARCTURUS PicoPure RNA Isolation Kit (ThermoFisher) and assayed on a 2100 Bioanalyzer with the RNA 6000 Pico Kit (Agilent) before amplification and cDNA synthesis with the Ovation RNA-Seq System V2 (NuGEN). Library preparation with **which kit** and 50-base, single-end sequencing on the HiSeq 2000 platform (Illumina) were performed by IGA Technology Services (Udine, Italy).

## Data analyses

Analysis of the sequenced libraries was carried out with `bash` and `R` scripts [@RALanguageandEn:2015wf], which were arranged into a pipeline using the `ruffus` package for `python3` [@Goodstadt:2010jp]. All of the code used for the analysis and to generate text, figures and tables for this report, along with the versions and paramaters of the software used, are available with revision history in a public GitHub repository at https://github.com/evoreprice/lmdPaper.

Briefly, adaptor trimming was performed with `cutadapt` [@Martin:2011eu] and reads were mapped against the MSUv7 genome and annotation [@Kawahara:2013gh; @Ouyang:2007il] downloaded from Phytozome 10.3 [@Goodstein:2012ho] using `STAR` [@Dobin:2012fg] in 2-pass mode. The `'quantMode'` argument of `STAR` was used for read counting. rRNA and tRNA contamination was estimated using `htseq-count` [@Anders:2014gf] to count the number of reads that mapped to regions annotated as tRNA or rRNA on the Rap-DB [@Ohyanagi:2006go] or to regions where reads simulated with `wgsim` (https://github.com/lh3/wgsim) from rRNA sequences in the TIGR Plant Repeats Database [@Ouyang:2004ia] also mapped. Differential expression analysis to calculate Log~2~-fold change (L~2~FC) values and transformed read counts was performed with the `DESeq2` package [@Love:2014do]. To determine a strict cut-off for unexpressed genes, expression values in transcripts per million were calculated for each gene [@Li:2010cl; @Wagner:2012jf]. These values were compared to pseudo-expression values calculated from intergenic regions of the genome, using intervals of a similar size distribution to those used for calculating the 'genic' expression values. For each library, the cut-off was placed at the 95^th^ percentile of the distribution of intergenic expression values. Transformed counts and L~2~FC values from expressed genes were used for soft clustering with the `Mfuzz` package [@Kumar:2007uw] and geneset enrichment analysis with the `gage` package [@Luo:2009kp] respectively. All plots were produced with the `ggplot2` package [@Wickham:2009tn].

### Comparative analysis of MADS family

The primary sequences of *Arabidopsis thaliana*, *Solanum lycopersicum* and *Oryza sativa* MADS proteins were downloaded by querying Phytozome 10.3 [@Goodstein:2012ho] for the PANTHER family PTHR11945 ('MADS BOX PROTEIN') via the `biomaRt` interface [@Thomas:2003db; @Durinck:2009ki]. An initial alignment of all proteins was performed with `Clustal Omega` using a full distance matrix [@Sievers:2011fn]. To clean the alignment, poorly conserved regions where greater than `r minpcnongap` percent of sequences had gaps or less than `r minpcident` percent of pairwise comparisons between sequences had identical residues were removed. To recover MADS genes related to the Arabidopsis MIKC clade, a UGPMA tree was estimated from the cleaned alignment and the tree was cut at the lowest height that returned a clade including all type II Arabidopsis MADS proteins listed in the Arabidopsis transcription factor database on the Arabidopsis Gene Regulatory Information Server [@Yilmaz:2011ew], with as few type I proteins as possible. Protein sequences in this clade that were longer than `r minProtLength` amino acids were realigned using `Clustal Omega`, and the alignment was cleaned as described above and used to estimate a neighbour-joining tree with the `bionj` algorithm implemented in the `ape` package [@Paradis:2004dv] rooted to type I MADS sequences (`r og`). L~2~FC change expression values for tomato and Arabidopsis genes were estimated from samples that have previously been described [@Mantegazza:2014di; @Park:2012dg] using the same method described above for the rice samples. The tree was plotted using the `ggtree` extension for `ggplot2` (**ref ggtree**).

# Accession Numbers

# Acknowledgements

# Supporting Information

**`r I(pasteLabel("Figure S", s_figCount, "sf_pca"))`**: Principal components analysis of transformed read counts for each library.

**`r I(pasteLabel("Figure S", s_figCount, "sf_isGenesTpm"))`**: Comparison of the LMD RNA-sequencing dataset with published *in situ* results.

**`r I(pasteLabel("Figure S", s_figCount, "sf_inSitu"))`**: Placeholder for the *in situ* results figure.

**`r I(pasteLabel("Figure S", s_figCount, "sf_mfuzzPcaCentroids"))`**: Assessment of clustering results.

**`r I(pasteLabel("Figure S", s_figCount, "sf_madsTree"))`**: Differential expression
of genes encoding putative MIKC proteins in *A. thaliana*, *Solanum lycopersicum* and *Oryza sativa* ssp. *japonica*.

**`r I(pasteLabel("Table S", s_tableCount, "st_libStats"))`**: Library and sequencing statistics.

**`r I(pasteLabel("Table S", s_tableCount, "st_refTable"))`**: References for `r I(pasteLabel("Figure S", s_figCount, "sf_isGenesTpm"))`.

**Data S1**: List of clustered genes with annotations.

# Tables

```{r t_hypergeom, results = "asis"}
pander::set.alignment('center', row.names = 'right')
pander::set.caption(paste0(
  I(pasteLabel("**Table ", tableCount, "t_hypergeom")),
  "**: Transcription factor and transcriptional regulator family enrichment for families that had at least two members in any one cluster, including the number of family members (*n*), number expressed in the whole dataset (*n*~expr~) and number and adjusted *p*-value for each cluster."))
pander::pandoc.table(t_hypergeom, split.tables = Inf, emphasize.rownames = FALSE)
```

# Figure legends

----

![](data/lmdFigure.pdf)

**`r I(pasteLabel("Figure ", figCount, "f_lmdFigure"))`**. Laser microdissection of meristem samples in the early stages of inflorescence development. Samples were collected from the rachis meristem (**a** and **e**), the primary branch meristem (**b** and **f**), the elongating primary branch meristem and secondary branch meristem (**c** and **g**) and the spikelet meristem (**d** and **h**). Images show the samples before (**a**–**d**) and after (**e**–**h**) dissection.

----

```{r f_mfuzzClusters, fig.height = 8.661 / 2}
print(f_mfuzzClusters)
```

**`r I(pasteLabel("Figure ", figCount, "f_mfuzzClusters"))`**. Common patterns of gene expression in developing inflorescences. Fuzzy *c*-means clustering of normalised, variance-stabilised read counts was used to recover `r as.character(english::english(c))` common patterns of expression. Each line describes the expression pattern of one gene, with the gene's membership to the cluster represented by the colour of the line. The core values for each cluster are plotted in black. The ordering of the panels in the plot was rearranged to enable side-by-side comparison of complementary clusters.

----

```{r f_gsea, fig.height = 8.661 / 2}
print(f_gsea)
```

**`r I(pasteLabel("Figure ", figCount, "f_gsea"))`**. Geneset enrichment analysis (GSEA) for transcription factor and other transcriptional regulator families. Test statistics were calculated from the L~2~FC values between the stage of interest and all other stages for the expressed genes in each geneset. The absolute value of the test statistic indicates the magnitude of the geneset-level change, and a positive test statistic indicates enrichment and a negative statistic indicates depletion. Although some test statistics had low *p*-values none were lower than 0.10 after Benjamini-Hochberg correction.

----

```{r f_alogFamily, fig.width = 6.614, fig.height = 6.614}
# set layout
grid::grid.newpage()
grid::pushViewport(grid::viewport(layout = grid::grid.layout(4, 4)))

# print the plots
print(f_alogFamily_a, vp = grid::viewport(layout.pos.row = 1:3, layout.pos.col = 1))
print(f_alogFamily_b, vp = grid::viewport(layout.pos.row = 1:3, layout.pos.col = 2))
print(f_alogFamily_c, vp = grid::viewport(layout.pos.row = 1:3, layout.pos.col = 3:4))
print(f_alogFamily_d, vp = grid::viewport(layout.pos.row = 4, layout.pos.col = 1:4))
```

**`r I(pasteLabel("Figure ", figCount, "f_alogFamily"))`**. Expression of selected *STMADS11*-like MADS genes (**a**), detected *G1*-Like (G1L/ALOG) genes (**b**), *SPL* genes and a microRNA precursor (**c**) and cytokinin-related genes (**d**) in inflorescence meristems in transcripts per million (TPM). A red point indicates that the gene was above the detection cutoff, and blue represents that the gene was not detected. Two reported *STMADS11*-like (*SVP*-like) targets of G1L5, *MADS22* and *MADS55* [@Yoshida:2013ff], are also more strongly expressed in the RM than in other meristems, but the other *STMADS11*-like gene, *MADS47*, was only detected in one library. Only three G1L genes (*G1L1*, *G1L2* and *G1L5*/*TAWAWA1*) were detected, but they share a similar pattern of expression. Several *SPL* genes and a co-regulated microRNA precursor, *MIR319A*, recovered from cluster `r getClusterName('small', 'dec')`, are highly expressed in RM and PBM before decreasing in expression in ePBM/SBM and SM. *GN1A* (*CKX2*) was not detected in the LMD dataset, but *LOG1* is expressed in all meristem types, and two other genes possibly related to cytokinin activation, *LABA1* (*LOGL6*) and *LOGL1*, are expressed more highly in ePBM/SBM and SM than in RM and PBM. 

----

```{r f_hb, fig.height = 8.661}
print(f_hb)
```

**`r I(pasteLabel("Figure ", figCount, "f_hb"))`**. Patterns of expression of homeobox genes in the meristem samples. The heatscale represents scaled, VST-transformed read counts for detected homeobox genes, and the coloured bar next to the *y*-axis shows the homeobox clade for each gene [@Jain:2008kg]. There is a set of homeobox genes that peak in expression in the RM, and another set that have their lowest expression in the RM and highest expression in PBM and SBM samples. The former group contains *OSH1* (**function?**), whilst the latter contains five of the eight expressed HD-ZIP IV homeobox genes.

----
# `sessionInfo` (*hide*)

```{r sessionInfo, results = "asis"}
cat(paste0("**branch:** _", system("git rev-parse --abbrev-ref HEAD", intern = TRUE), "_\n\n"))
cat(paste0("**hash:** _", system("git rev-parse HEAD", intern = TRUE), "_\n\n"))
pander::pander(sessionInfo())
```

# References (move to before **Tables**)
