---
title: Genome-wide changes in expression during inflorescence architecture development
  in rice
csl: csl/the-plant-journal.csl
header-includes: \usepackage{graphicx}
output:
  BiocStyle::html_document:
    fig_caption: yes
    force_captions: yes
    highlight: pygments
    keep_md: yes
    number_sections: yes
    dev: svg
  word_document:
    fig_caption: yes
  pdf_document:
    fig_caption: yes
    keep_tex: yes
    number_sections: yes
bibliography: bib/test.bib
---

```{r Setup, include=FALSE, results="hide", warning=FALSE}

knitr::opts_chunk$set(fig.path='figures/',
               cache.path='cache/',
               dev=c("svg", "cairo_pdf"),
               fig.width=3.150,
               fig.height=6.3,
               fig.align = 'center',
               dpi=300,
               fig.show='hold',
               cache=FALSE,
               echo=FALSE,
               results="hide",
               message=FALSE,
               warning=FALSE)

# https://gist.githubusercontent.com/rmflight/3858973/raw/591ee603ce0996ac407902f7dd59520b35b3702a/tableFigureFuncs.r
pasteLabel <- function(preText, inObj, objName, insLink=TRUE){
	objNum <- inObj[objName]
	
	useText <- paste(preText, objNum, sep=" ")
	if (insLink){
		useText <- paste("[", useText, "](#", objName, ")", sep="")
	}
	useText
}

# this increments the counter, and gives a name to the number so we can reference it later
incCount <- function(inObj, useName){
	nObj <- length(inObj)
	useNum <- max(inObj) + 1
	inObj <- c(inObj, useNum)
	names(inObj)[nObj+1] <- useName
	inObj
}
figCount <- c("_"=0)
tableCount <- c("_"=0)

tableCat <- function(inFrame){
	outText <- paste(names(inFrame), collapse=" | ")
	outText <- c(outText, paste(rep("---", ncol(inFrame)), collapse=" | "))
	invisible(apply(inFrame, 1, function(inRow){
		outText <<- c(outText, paste(inRow, collapse=" | "))
	}))
	return(outText)
}

capwords <- function(s, strict = FALSE) {
    cap <- function(s) paste(toupper(substring(s, 1, 1)),
                  {s <- substring(s, 2); if(strict) tolower(s) else s},
                             sep = "", collapse = " " )
    sapply(strsplit(s, split = " "), cap, USE.NAMES = !is.null(names(s)))
}

source('src/R/figures.R')

```

Authors:

Addresses:

### Running title

### Keywords

### SRA details for libaries

Word count

***

# Summary

# Significance statement

# Introduction

* Rice inflorescence development
+ branching & grain yield
+ sequential changes in meristem activity
+ few key genes

# Results and Discussion

```{r lmdFigure}
figCount <- incCount(figCount, "f_lmdFigure")
f_lmdFigure <- "data/lmdFigure.pdf"
```

## Dissections in early stages of inflorescence development

+  **Figure**: laser microdissections

To investigate gene expression during inflorescence development, LASER microdissection (LMD) was used to collect meristematic tissues from the rachis meristem (RM), primary branch meristem (PBM), elongating primary branch meristem and secondary branch meristem (ePBM/SBM) and spikelet meristem (SM) (`r I(pasteLabel("Figure", figCount, "f_lmdFigure"))`). RNA from the meristem samples was isolated, amplified and used for the production of a sequencing library. **Details of the procedure, yield, etc.**. 

<a id="f_lmdFigure"></a>
![](data/lmdFigure.pdf)

## Transcriptome sequencing of meristem samples

+ Max. 1 paragraph about boring sequencing stats
+ **Table** (or supplementary): # reads, mapping %, rRNA, introns, genes blah blah
+ Description of the dataset


## Clustering of expression patterns

```{r mfuzz}
figCount <- incCount(figCount, "f_mfuzzClusters")
tableCount <- incCount(tableCount, "t_hypergeom")
t_hypergeom <- readRDS('output/mfuzz/sigFamilies.Rds')
# fix p-values
t_hypergeom[,grepl("^\\*p", colnames(t_hypergeom))] <- gsub("e-(\\d+)", "x10^-\\1^",
format.pval(t_hypergeom[,grepl("^\\*p", colnames(t_hypergeom))], digits = 2,
            na.form = ""))
t_hypergeom[is.na(t_hypergeom)] <- ""
```

To recover common expression patterns in the meristem samples, read counts were transformed using the variance-stabilizing transformation (VST) included in the `DESeq2` package [@Love:2014do]. After determining an expression cutoff for each library, genes that were not above the cutoff in any sample were excluded, leaving (**how many genes remaining, link to methods**). These genes were ranked by variance, and genes with variance higher than the 90th percentile were used for clustering (**how many genes**). The standardised, VST-transformed read counts for the expressed, variable genes were clustered using the fuzzy *c*-means algorithm implemented in the `Mfuzz` package [@Futschik:2005vo]. To determine the number of cluster cores (*c*), the minimum distance between cluster centroids, the formation of empty clusters, and PCA plots of cluster members were monitored for clusters produced with *c* values between 2 and `r maxClust`. A cluster number of `r as.character(english::english(c))` *justify chosing this? double diffential was closest to zero?* (**`r I(pasteLabel("Figure", figCount, "f_mfuzzClusters"))`**).

<a id="f_mfuzzClusters"></a>
```{r f_mfuzzClusters, fig.height=6.3}
print(f_mfuzzClusters)
```

After examining the lists of genes in each cluster it was evident that most clusters had an enrichment of transcription factors (**tables in SI**). To determine whether particular TF families were associated with particular patterns of expression, the enrichment of TF family members in each cluster was calculated using hypergeometric tests. These calculations were carried out for families that had two or more members in any given cluster (**`r I(pasteLabel("Table", tableCount, "t_hypergeom"))`**). After adjustment for multiple comparisons,   MADS genes were significantly over-represented in clusters 1 and 4, both of which contain genes that have higher expression in later meristem samples. ABI3VP transcription factors are 

<a id="t_hypergeom"></a>
```{r hypergeom, results = "markup"}
knitr::kable(t_hypergeom, caption = 'Transcription factor family enrichment per cluster', align = 'r')
```

+ Clusters including 'known' genes

## Patterns of expression of transcription families

+ **Figure**: GSEA of transcription factor families

Because of the enrichment of TF family members in some of the gene expression clusters, geneset-level enrichment analysis (GSEA) was used to compare the levels of expression of TF family members in the meristem samples. **Something like: Consistent with their enrichment in cluster 1 and 6, MADS genes were significantly enriched in the SM/FM samples**. TF families involved in branching were enriched in **which families, which samples**.

## MADS genes: comparison of expression & homology in rice, tomato, arabidopsis

The presence of MADS genes was enriched in **which clusters** and their overall expression is enriched in the spikelet meristem sample. These results correspond with previous studies that have established the importance of MADS genes in inflorescence development. The evolution of their expression patterns was 

+ **Figure**: Similarity tree of MADS proteins with expression values and dN/dS values for each one.

# Experimental procedures

# Accession Numbers

# Acknowledgements

# Short legends for Supporting Information

# References

***

# Tables

***

# Figure legends

`r I(pasteLabel("Figure", figCount, "f_lmdFigure"))`
**LASER microdissection of meristem samples in the early stages of inflorescences development.** Samples were collected from the rachis meristem (**a** and **e**), the primary branch meristem (**b** and **f**), the elongating primary branch meristem and secondary branch meristem (**c** and **g**) and the spikelet meristem (**d** and **h**). Images show the samples before (**a**–**d**) and after (**e**–**h**) dissection.

`r I(pasteLabel("Figure", figCount, "f_mfuzzClusters"))`
**Common patterns of expression in the meristem samples.** Fuzzy *c*-means clustering of normalised, variance-stabilised read counts was used to recover `r capwords(as.character(english::english(c)))` common patterns of expression. Each line describes the expression pattern of one gene, with the gene's membership to the cluster mapped to the colour of the line. The core values for each cluster are plotted in black. 

***

# Figures

***

# Junk

I think `inline code` should work?