---
title: Genome-wide changes in expression during inflorescence architecture development
  in rice
csl: csl/the-plant-journal.csl
header-includes: \usepackage{graphicx}
output:
  BiocStyle::html_document:
    fig_caption: yes
    force_captions: yes
    highlight: pygments
    keep_md: yes
    number_sections: yes
    dev: "CairoPDF"
  word_document:
    fig_caption: yes
    reference_docx: csl/wordStyles.docx
  pdf_document:
    fig_caption: yes
    keep_tex: yes
    number_sections: yes
bibliography: bib/papersLib.bibtex
---

```{r Setup, include=FALSE, results="hide", warning=FALSE}

knitr::opts_chunk$set(fig.path='figures/',
               cache.path='cache/',
               dev=c("png", "cairo_pdf"),
               fig.width=3.150,
               fig.height=8.661,
               fig.align = 'center',
               dpi=300,
               fig.show='hold',
               cache=FALSE,
               echo=FALSE,
               results="hide",
               message=FALSE,
               warning=FALSE)

pander::panderOptions("table.caption.prefix", "")

source('src/R/figures.R')

```

Thomas W.R. Harrop^§^[^1], Israr Ud Din^¶^[^1], Veronica Gregis^¶^, Michela Osnato^¶^, Stefan Jouannic^§^, Hélène Adam^§^ and Martin Kater^¶^

[^1]: These authors contributed equally to this work.

| ^§^ Institut de Recherche pour le Développement (IRD), 911 Avenue Agropolis, 34394 Montpellier, France.
| ^¶^ Università degli Studi di Milano, Via Celoria 26, 20133 Milano, Italy.

## Running title

## Keywords

## SRA details for libaries

## Word count

# Summary

# Significance statement

# Introduction

* Rice inflorescence development
+ Branching & grain yield
+ Sequential changes in meristem activity
+ Discuss a few key genes

# Results and Discussion

To investigate gene expression during inflorescence development, LASER microdissection (LMD) was used to collect meristematic tissues from the rachis meristem (RM), primary branch meristem (PBM), elongating primary branch meristem and secondary branch meristem (ePBM/SBM) and spikelet meristem (SM) (`r I(pasteLabel("Figure ", figCount, "f_lmdFigure"))`). RNA isolated from the meristem samples was amplified and used to produce cDNA libraries for sequencing. **Need area cut +/- number of panicles used for each library from Israr**. RNA input for library preparation was between `r min(st_libStats$"Yield (ng)")` and `r max(st_libStats$"Yield (ng)")` ng and the average RNA integrity number was `r round(mean(st_libStats$"RIN"), 1)` (`r I(pasteLabel("Table S", s_tableCount, "st_libStats"))`). Initially, two biological replicates were prepared for each stage, but after performing a principal components analysis (PCA) on transformed expression values, two libraries that were produced from degraded RNA were excluded from the analysis (`r I(pasteLabel("Figure S", s_figCount, "sf_pca"))`). Following this, further samples were prepared to provide a total of three biological replicates with intact RNA (RIN ≥ 7) for each meristem type.

![](data/lmdFigure.pdf)

At least `r round(min(st_libStats$"Reads (M)"), 0)` million 50-base reads were produced from each library, yielding between `r round(min(st_libStats$"Reads in genes (M)", na.rm = TRUE), 0)` and `r round(max(st_libStats$"Reads in genes (M)", na.rm = TRUE), 0)` million reads that mapped to genes after excluding reads likely to have originated from rRNA and tRNA. Using strict cutoffs for gene expression, `r length(unique(readRDS("output/expressedGenes/expressedGenesAll.Rds")))` unique genes were detected in at least one of the libraries. As a preliminary quality check, the qualitative and quantitative expression of a selection of genes were compared with patterns that had previously been reported by *in situ* hybridisation (`r I(pasteLabel("Table S", s_tableCount, "st_reviewInSitu"))` and `r I(pasteLabel("Figure S", s_figCount, "sf_isGenesTpm"))`).

**Our in situ results could go here (or in SI).**

##  Patterns of gene expression during inflorescence development

To recover common expression patterns in expressed genes, read counts were transformed using the variance-stabilizing transformation (VST) included in the `DESeq2` package [@Love:2014do]. Expressed genes were ranked by variance, and standardised, VST-transformed read counts for the 25% most-variable genes were clustered using the fuzzy *c*-means algorithm implemented in the `Mfuzz` package [@Kumar:2007uw]. To determine the number of cluster cores (*c*), the minimum distance between cluster centroids, the formation of empty clusters, and PCA plots of cluster members were monitored for clusters produced with *c* values between 2 and `r maxClust` (`r I(pasteLabel("Figures S", s_figCount, "sf_mfuzzCentroids"))` `r I(pasteLabel(" and S", s_figCount, "sf_mfuzzPca"))`), leading to the use of `r as.character(english::english(c))` cluster cores (`r I(pasteLabel("Figure ", figCount, "f_mfuzzClusters"))`). There are several expression patterns descibed by the clusters: genes that increase or decrease steadily during development (clusters `r getClusterName('inc', 'inc')` and `r getClusterName('dec', 'dec')`), genes that change expression between RM and PBM and remain constant from PBM to SM (clusters `r getClusterName('inc', 'small')` and `r getClusterName('dec', 'small')`), genes that are constant between RM and PBM and then change between PBM and SM (clusters `r getClusterName('small', 'inc')` and `r getClusterName('small', 'dec')`). There were also two weaker clusters containing fewer genes, which had complex expression patterns involving changes in expression particular to the PBM (clusters `r getClusterName('inc', 'dec')` and `r getClusterName('dec', 'inc')`).

```{r f_mfuzzClusters, fig.height = 8.661 / 2}
print(f_mfuzzClusters)
```

Examination of the clustered genes suggested that transcription factors were overrepresented in most clusters (**annotated cluster tables in SI**). To determine whether particular TF families were associated with particular patterns of expression, the enrichment of TF family members in each cluster was calculated using hypergeometric tests. Using the list of transcription factor and other transcriptional regulator families available on the Plant Transcription Factor Database [@PerezRodriguez:2010co], a hypergeometric test was carried out if there were at least two members of one family in a given cluster. After adjustment for multiple comparisons, each cluster had a significant enrichment of at least one family (`r I(pasteLabel("Table ", tableCount, "t_hypergeom"))`). Cluster `r getClusterName('inc', 'small')`, containing genes that are more highly expressed in the PBM–SM samples than in the RM sample, has an enrichment of Homeobox and basic helix-loop-helix TFs whereas the cluster with the complementary pattern (cluster `r getClusterName('dec', 'small')`) is enriched for ABI3VP1 and G2-like TFs. MADS TFs and TRAF genes were overreprsented among genes that have low expression in the SM and PBM but higher expression in the RM (cluster `r getClusterName('small', 'inc')`) and bHLH, SBP and C2C2-GATA TFs were overreprented among genes with the inverse pattern (cluster `r getClusterName('small', 'dec')`)

|**Discussion: what might these patterns mean biologically?**
|**Discussion: which clusters have genes that are already candidates for panicle branching?**

## Dynamic expression of transcription factor families

Because of the enrichment of TF family members in some of the gene expression clusters, geneset enrichment analysis (GSEA) was used to provide an overview of expression of transcription factors and other regulators in the meristem samples (`r I(pasteLabel("Figure ", figCount, "f_gsea"))`). Unlike soft clustering, which groups genes based on shared expression dynamics, this GSEA was used to investigate the overall strength of expression of the families based on Log~2~-fold changes (L~2~FC) in the expression of their members in each stage. These patterns are not intended to represent each member of each gene family: for example, there are several MADS genes that have lower expression in the SM than in the RM, but the test statistic for the whole MADS family was higher in the SM than in the RM. Despite this caveat, there are some notable modes of expression shared between different families that may indicate roles in inflorescence development. The most pronounced change in expression appears to occur between the RM and branch meristem samples. Several families (including TUBBY-like, GROWTH-REGULATING FACTOR, bZIP and ABI3VP1 TF genes and Aux/IAA genes) are strongly expressed in RM and lower in the other meristem types. Another group, including mTERF, MYB, SBP and FHA TF genes and SET, PHD and Jumonji regulator genes, is enriched in one or both of the PBM and ePBM/SBM samples but depleted or not enriched in the RM samples.

```{r f_gsea, fig.height = 8.661 / 2}
print(f_gsea)
```

**Discussion: In which samples were TF families involved in branching enriched?**

## MADS genes: comparison of expression & homology in rice, tomato, arabidopsis

There was an overrepresentation of MADS genes in clusters **xyz** and their overall expression is enriched in the spikelet meristem sample. These results correspond with previous studies that have established the importance of MADS genes in inflorescence development in rice and other plants (**expand**).

To investigate the evolution of the MADS family in Angiosperms, their expression values in the dicots *A. thaliana* and *Solanum lycopersicum* and the monocot *O. sativa* were compared to their homology in these species. 

* **Figure**: Similarity tree of MADS proteins with L2FC values between comparable samples (and dN/dS values?) for each one:
+ **rice**: rachis meristem vs. spikelet merisem.
+ **arapidospis**: inflorescence meristem vs. floret meristem.
+ **tomato**: simpodial meristem vs. floret meristem. (Need to find list of tomato MADS genes. List of ~32 MIKC domain proteins on the other plant TFDB, http://planttfdb.cbi.pku.edu.cn/)
+ (could I map the l2fc to the background of the geom_label rectangle? "The fill aesthetic controls the background colour of the label.")

# Experimental procedures

## Plant material and sampling

*Oryza sativa* ssp. *japonica* cv. Nipponbare plants were grown in a growth chamber with 70% relative humidity at 30°C during the day and 26°C at night. After 7 weeks in long day conditions (LD; 14 hours light and 10 hours darkness) plants were transferred to short day conditions (SD; 10 hours light and 14 hours darkness) to induce floral transition. Inflorescence meristems were harvested from 6 to 14 days after the change of photoperiod.

## Tissue embedding, LASER microdissection and sequencing

Each inflorescence was harvested in 2 mL ice-cold 3:1 ethanol:acetic acid fixative solution, infiltrated (=**filtered**?) twice under a mild vacuum for 15 minutes and stored for 20 hours in fresh fixative at 4°C. Embedding for dissection was performed as described by @Mantegazza:2014di. Tissues were cut into 8 µm sections on an RM2155 or RM2255 microtome (Leica) and dissected on a LMD6000 or LMD7000 LASER dissector (Leica). RNA was amplified using the ARCTURUS PicoPure RNA Isolation Kit (ThermoFisher) and assayed on a 2100 Bioanalyzer with the RNA 6000 Pico Kit (Agilent) before amplification and cDNA synthesis with the Ovation RNA-Seq System V2 (NuGEN). Library preparation with **which kit** and 50-base single-end sequencing on the HiSeq 2000 platform (Illumina) were performed by IGA Technology Services (Udine, Italy).

## Data analyses

Analysis of the sequenced libraries was carried out with `bash` and `R` scripts [@RALanguageandEn:2015wf], which were arranged into a pipeline using the `ruffus` package for `python3` [@Goodstadt:2010jp]. All of the code used for the analysis and to generate text, figures and tables for this report, along with the versions and paramaters of the software used, are available with revision history in a public GitHub repository at https://github.com/evoreprice/lmdPaper.

Briefly, adaptor trimming was performed with `cutadapt` [@Martin:2011eu] and reads were mapped against the MSUv7 genome and annotation [@Kawahara:2013gh; @Ouyang:2007il] downloaded from Phytozome 10.3 [@Goodstein:2012ho] using `STAR` [@Dobin:2012fg] in 2-pass mode. The `'quantMode'` argument of `STAR` was used for read counting. rRNA and tRNA contamination was estimated using `htseq-count` [@Anders:2014gf] to count the number of reads that mapped to regions annotated as tRNA or rRNA on the Rap-DB [@Ohyanagi:2006go] or to regions where reads simulated with `wgsim` (https://github.com/lh3/wgsim) from rRNA sequences in the TIGR Plant Repeats Database [@Ouyang:2004ia] also mapped. Differential expression analysis to calculate Log~2~-fold change (L~2~FC) values and transformed read counts was performed with the `DESeq2` package [@Love:2014do]. To determine a strict cutoff for unexpressed genes, expression values in transcripts per million were calculated for each gene [@Li:2010cl; @Wagner:2012jf]. These values were compared to pseudo-expression values calculated from intergenic regions of the genome, using intervals of a similar size distrubution to those used for calculating the 'genic' expression values. For each library, the cutoff was placed at the 95^th^ percentile of the distribution of intergenic expression values. Transformed counts and L~2~FC values from expressed genes were used for soft clustering with the `Mfuzz` package [@Kumar:2007uw] and geneset enrichment analysis with the `gage` package [@Luo:2009kp] respectively. All plots were produced with the `ggplot2` package [@Wickham:2009tn].

### Comparative analysis of MADS family

The primary sequences of *Arabidopsis thaliana*, *Solanum lycopersicum* and *Oryza sativa* MADS proteins were downloaded by querying Phytozome 10.3 [@Goodstein:2012ho] for the PANTHER family PTHR11945 ('MADS BOX PROTEIN') via the `biomaRt` interface [@Thomas:2003db, @Durinck:2009ki].

# Accession Numbers

# Acknowledgements

# Short legends for Supporting Information

**`r I(pasteLabel("Table S", s_tableCount, "st_libStats"))`**: Library and sequencing statistics.

**`r I(pasteLabel("Table S", s_tableCount, "st_reviewInSitu"))`**: Comparison of the dataset with published *in situ* expression profiles.

**`r I(pasteLabel("Figure S", s_figCount, "sf_pca"))`**: Principal components analysis of transformed read counts for each library.

**`r I(pasteLabel("Figure S", s_figCount, "sf_isGenesTpm"))`**: Quantitative expression values for the genes that were compared to published *in situ* results.

**`r I(pasteLabel("Figure S", s_figCount, "sf_mfuzzCentroids"))`**: Minimum centroid distance for 2–`r maxClust` cluster cores (*c*).

**`r I(pasteLabel("Figure S", s_figCount, "sf_mfuzzPca"))`**: Classical multidimensional scaling (MDS) of Euclidean distances between transformed read counts for clustered genes.

# Tables

```{r t_familiesPerCluster, results = "asis"}

pander::set.caption(paste0(
  I(pasteLabel("**Table ", tableCount, "t_familiesPerCluster")),
  "**: Transcription factor and transcriptional regulator family members in each cluster. *n*, total number of family members, *n*~expr~, number of family members expressed in the whole dataset, C~1~–C~", c, "~, number of family members in each cluster, and the total number of family members in all clusters."))
t_familiesPerCluster <- as.data.frame(t_familiesPerCluster[!Total == 0])
rownames(t_familiesPerCluster) <- t_familiesPerCluster$Family
t_familiesPerCluster$Family <- NULL
pander::pandoc.table(t_familiesPerCluster, split.tables = Inf, emphasize.rownames = FALSE,
                     justify = c("right", rep("centre", length(names(t_familiesPerCluster)) - 1), "left"))

```


```{r t_hypergeom, results = "asis"}
pander::set.alignment('center', row.names = 'right')
pander::set.caption(paste0(
  I(pasteLabel("**Table ", tableCount, "t_hypergeom")),
  "**: Transcription factor and transcriptional regulator family enrichment for families that had at least two members in any one cluster, including the number of family members (*n*), number expressed in the whole dataset (*n*~expr~) and number and adjusted *p*-value for each cluster."))
pander::pandoc.table(t_hypergeom, split.tables = Inf, emphasize.rownames = FALSE)
```

# Figure legends

**`r I(pasteLabel("Figure ", figCount, "f_lmdFigure"))`**. LASER microdissection of meristem samples in the early stages of inflorescence development. Samples were collected from the rachis meristem (**a** and **e**), the primary branch meristem (**b** and **f**), the elongating primary branch meristem and secondary branch meristem (**c** and **g**) and the spikelet meristem (**d** and **h**). Images show the samples before (**a**–**d**) and after (**e**–**h**) dissection.

**`r I(pasteLabel("Figure ", figCount, "f_mfuzzClusters"))`**. Common patterns of gene expression in developing inflorescences. Fuzzy *c*-means clustering of normalised, variance-stabilised read counts was used to recover `r as.character(english::english(c))` common patterns of expression. Each line describes the expression pattern of one gene, with the gene's membership to the cluster represented by the colour of the line. The core values for each cluster are plotted in black. The ordering of the panels in the plot was rearranged to enable side-by-side comparison of complementary clusters.

**`r I(pasteLabel("Figure ", figCount, "f_gsea"))`**. Geneset enrichment analysis (GSEA) for transcription factor and other transcriptional regulator families. Test statistics were calculated from the L~2~FC values between the stage of interest and all other stages for the expressed genes in each geneset. The absolute value of the test statistic indicates the magnitude of the geneset-level change, and a positive test statistic indicates enrichment and a negative statistic indicates depletion. Although some test statistics had low *p*-values none were lower than 0.10 after Benjamini-Hochberg correction.

# `R sessionInfo()` (*hide*)

```{r sessionInfo, results = "asis"}
pander::pander(sessionInfo())
git <- c(paste("**git branch:**",system("git rev-parse --abbrev-ref HEAD", intern = TRUE)),
          paste("**git hash:**", system("git rev-parse HEAD", intern = TRUE)))

for (x in git){cat(x, "\n")}

```

# References (move to before *Tables*)
