---
title: Gene expression profiling of reproductive meristem types in early rice inflorescences by laser microdissection
csl: csl/the-plant-journal.csl
header-includes:
  \usepackage{graphicx}
  \usepackage[utf8]{inputenc}
  \DeclareUnicodeCharacter{2265}{$\geq$}
output:
  word_document:
    fig_caption: yes
    reference_docx: csl/wordStyles.docx
  BiocStyle::html_document:
    dev: CairoPDF
    fig_caption: yes
    force_captions: yes
    highlight: pygments
    number_sections: yes
bibliography: bib/papersLib.bibtex
---

```{r Setup, include=FALSE, results="hide", warning=FALSE}

# max width (2 col): 6.614
# max width (1 col): 3.150
# max height : 8.661

knitr::opts_chunk$set(fig.path='figures/',
               cache.path='cache/',
               dev=c("png", "pdf"),
               dev.args=list(
                 pdf = list(colormodel = 'cmyk'),
                 png = list()),
               fig.width=3.150,
               fig.height=8.661,
               fig.align = 'center',
               dpi=300,
               fig.show='hold',
               cache=FALSE,
               echo=FALSE,
               results="hide",
               message=FALSE,
               warning=FALSE)

pander::panderOptions("table.caption.prefix", "")

source('src/R/figures.R')

```

Thomas W.R. Harrop^§,1^, Israr Ud Din^¶,†,^[^1], Veronica Gregis^¶^, Michela Osnato^¶,‡^, Stefan Jouannic^§^, Hélène Adam^§,2^ and Martin Kater^¶,†,^[^2]

[^1]: These authors contributed equally to this work.
[^2]: To whom correspondence should be addressed: martin.kater@unimi.it or helene.adam@ird.fr.

| ^§^ Institut de Recherche pour le Développement, UMR DIADE, 911 Avenue Agropolis, 34394 Montpellier, France.
| ^¶^ Dipartimento di Bioscienze, Università degli Studi di Milano, Via Celoria 26, 20133 Milano, Italy.
| ^†^ Dipartimento di Scienze Farmacologiche e Biomolecolari, Università degli Studi di Milano, Via Celoria 26, 20133 Milano, Italy.
| ^‡^ Present address: Centre for Research in Agricultural Genomics, Campus UAB, 08193 Cerdanyola, Spain.

**Running title**: Expression profile of rice inflorescence meristems

**Keywords**: Inflorescence development, rice, reproductive meristems, RNA sequencing

**Data availability**: Sequence data from this article have been deposited with the National Center for Biotechnology Information Sequence Read Archive under accession number SRP067488.

**Word count**:

# Summary

In rice, inflorescence architecture is established at early stages of reproductive development and contributes directly to grain yield potential. After induction of flowering, the complexity of branching and therefore the number of seeds on the panicle are determined by the activity of different meristem types and the timing of transitions between them. Although some of the genes involved in these transitions have been identified, understanding of the network of transcriptional regulators controlling this process is lacking. To address this, we used a precise laser microdissection and RNA sequencing approach in *Oryza sativa* ssp. *japonica* cv. Nipponbare to produce quantitative data that describe the landscape of gene expression in four different meristem types: the rachis meristem, the primary branch meristem, the elongating primary branch meristem (including axillary meristems) and the spikelet meristem. A switch in expression profile between apical and axillary meristem types followed by more gradual changes during transitions in axillary meristem identity was observed, and a number of genes potentially involved in branching were identified. This resource will be vital for a mechanistic understanding of the link between inflorescence development and grain yield.

# Significance statement

Rice inflorescence meristems undergo a series of transitions during differentiation, and control of these transitions impacts the morphology of the adult inflorescence and hence grain yield. We use laser microdissection and RNA sequencing to describe the transcriptional changes that occur during this process.

# Introduction

Agricultural development is essential to ensure food production and security for a growing population [@Borlaug:2007cp]. Rice is a staple food for over half the world's population, including many developing countries. A sustainable increase in the production of rice under the constraints of a changing climate and diminishing water and land availability will require plants with improved grain output, making the establishment of high-yield rice varieties a goal of modern breeding programs [@Peng:2008jb]. Rice yield is a complex trait influenced by genetic and epigenetic factors, and is progressively defined during the life cycle of the plant, first during the vegetative phase, where the number of fertile tillers is established, and then during the reproductive phase and grain-filling phase [@Ikeda:2004tk].

The branched inflorescence of rice is a compound raceme, classified as a panicle, and is composed of a rachis (the main axis), primary branches, higher order branches, and spikelets. The number of seeds on the panicle is influenced by the complexity and arrangement of branches and spikelets. The establishement and activity of apical and axillary meristems conditions branching in two phases during the early stages of panicle development: the timing of rachis meristem abortion determines the number of primary branches, whilst the transition of indeterminate branch meristems to determinate spikelet meristems specifies the complexity of branching. After spikelet differentiation, branching complexity is fixed and the rachis and branches elongate rapidly and heading and flowering occurs [@Ikeda:2004tk]. The landscape of gene expression is a characteristic that differentiates meristem types and is involved in the control of meristem identity transitions. Mutant analyses and mapping of quantitative trait loci have identified a number of genes required for the initiation and development of panicles, as well as genes controlling the number and size of grains and panicles [@Xing:2010de; @Wang:2011kq]. Some of these genes are involved in the patterning of axillary meristems and panicle branching, such as the nuclear regulatory factor-encoding genes *MONOCULM 1*, *LAX PANICLE 1* and *LAX PANICLE 2* [@Wang:2011kq]. A large set of genes related to floral development, which may also affect panicle architecture, has been identified [@Yoshida:2011bo], some of which may have been under selection during domestication or are associated with crop improvement in Asian rice  [@Xing:2010de; @He:2011io; @Wang:2011kq; @Xu:2012hq; @Ikeda:2013go].

Understanding the events during development that determine panicle characteristics such as branching complexity and its plasticity will be vital for sustainable improvement of rice yield potential using targeted breeding programs. Despite advances in the characterization of individual genes and their interactions, a complete understanding of the control of panicle morphology and grain yield will require mechanistic studies explaining the interactions of gene regulatory network components with each other and with the environment [@Azpeitia:2013jia]. One step towards this goal is to describe the differences in gene expression between different meristem types during development. In this article, we describe the measurement and analysis of genome-wide expression in meristematic tissues from the early stages of panicle development in *Oryza sativa* ssp. *japonica* cv. Nipponbare, using a precise laser microdissection and RNA sequencing approach.

# Results

## Panicle morphology and sampling for laser microdissection

In rice, the start of the reproductive phase and subsequent initiation of inflorescence development involves fast morphological transformations [described by @Ikeda:2004tk]. A detailed histological analysis of rice inflorescences was performed to select four morphologically distinct inflorescence meristem types for laser microdissection (LMD). After differentiation of the flag leaf, the first stage of reproductive development is the conversion of the shoot apical meristem (SAM) to the rachis meristem (RM). The first bract primordium was produced opposite the flag leaf (`r I(pasteLabel("Figure ", figCount, "f_lmdFigure"))`a). After the establishment of the RM, some cells differentiate into primary branch meristem (PBM) in the axils of newly developed bracts (`r I(pasteLabel("Figure ", figCount, "f_lmdFigure"))`b). After the formation of primary branches, bract growth ceased and primary branches elongated (ePBM) (`r I(pasteLabel("Figure ", figCount, "f_lmdFigure"))`c). During primary branch elongation, the PBM can give rise to axillary meristems (AM), which may differentiate into secondary and higher-order branches or be converted directly to spikelet meristem (SM). For this analysis, SM differentiation was considered as the final stage of panicle development (`r I(pasteLabel("Figure ", figCount, "f_lmdFigure"))`d). The PBM and secondary branch meristem (SBM) are both converted to a terminal SM. Each terminal or axillary SM produces one floret meristem (FM), which differentiates into a single floret. The RM, PBM and ePBM/AM stages are the indeterminate stages in which meristematic cells are maintained, whilst the SM has a determinate fate in which the stem cell activity will be lost and from which florets will differentiate [@Ikeda:2004tk].

## Genome-wide expression analysis of inflorescence development in rice

To investigate gene expression during the development of the rice inflorescence, LMD was used to collect meristematic tissues from the RM, PBM, ePBM/AM and SM of early panicles (`r I(pasteLabel("Figure ", figCount, "f_lmdFigure"))`e–l). RNA isolated from the meristem samples was amplified and used to produce cDNA libraries for sequencing. RNA input for amplification was between `r min(st_libStats$"Yield (ng)")` and `r max(st_libStats$"Yield (ng)")` ng and the average RNA integrity number (RIN) was `r round(mean(st_libStats$"RIN"), 1)` (`r I(pasteLabel("Table S", s_tableCount, "st_libStats"))`). Initially, two biological replicates were prepared for each stage, but after performing a principal components analysis (PCA) on transformed expression values, two libraries that were produced from degraded RNA (RIN < 6.5) were excluded from the analysis (`r I(pasteLabel("Figure S", s_figCount, "sf_pca"))`). Following this, further samples were prepared to provide a total of three biological replicates with intact RNA (RIN ≥ 7) for each meristem type. A single sequencing library was produced from each biological replicate.

At least `r round(min(st_libStats$"Reads (M)"), 0)` million single-end, 50-base reads were produced from each library, yielding between `r round(min(st_libStats$"Reads in genes (M)", na.rm = TRUE), 0)` and `r round(max(st_libStats$"Reads in genes (M)", na.rm = TRUE), 0)` million uniquely mapped reads within genes, after excluding reads likely to have originated from rRNA and tRNA. Using strict cut-offs for gene expression, `r length(unique(readRDS("output/expressedGenes/expressedGenesAll.Rds")))` unique genes were detected in two or more libraries from at least one stage (Data S1). As a preliminary assessment of the RNA sequencing (RNAseq) results, the qualitative and quantitative expression of `r nIS` genes were compared with previously reported patterns. `r nPos` of the `r nIS` genes were detected in two or more libraries from at least one meristem type in the LMD dataset (`r I(pasteLabel("Figure S", s_figCount, "sf_isGenesTpm"))` and `r I(pasteLabel("Table S", s_tableCount, "st_refTable"))`). The expression of a further four uncharacterized genes that were detected in specific meristem types by RNAseq was confirmed by RNA *in situ* hybridization. Each of the four genes was detected in the same meristem types by both methods (`r I(pasteLabel("Figure ", figCount, "f_inSitu"))`). These results suggest that the tissues used for RNA sequencing were accurately dissected and represent the intended meristem types.

##  Patterns of gene expression during inflorescence development

To recover common expression patterns, read counts were transformed using the variance-stabilizing transformation (VST) included in the `DESeq2` software package [@Love:2014do]. Detected genes were ranked by variance, and standardized, VST-transformed read counts for the `r as.character(dim(expressionMatrix)[1])` genes with the highest variance (33% of expressed genes) were clustered using the fuzzy *c*-means algorithm implemented in the `Mfuzz` package [@Kumar:2007uw]. To determine the number of cluster cores (*c*), the minimum distance between cluster centroids, the formation of empty clusters, and PCA plots of cluster members were monitored for clusters produced with *c* values between 2 and `r maxClust` (`r I(pasteLabel("Figure S", s_figCount, "sf_mfuzzPcaCentroids"))`), leading to the use of `r as.character(english::english(c))` cluster cores (`r I(pasteLabel("Figure ", figCount, "f_mfuzzClusters"))`). Several common patterns were recovered by clustering: genes that increase or decrease in expression steadily during development (clusters `r getClusterName('inc', 'inc')` and `r getClusterName('dec', 'dec')`), genes that change expression between the apical meristem (RM) and axillary meristem (PBM to SM) samples (clusters `r getClusterName('inc', 'small')` and `r getClusterName('dec', 'small')`) and genes that change expression gradually over the course of changes in axillary meristem identity (clusters `r getClusterName('small', 'inc')` and `r getClusterName('small', 'dec')`). There were also two weaker clusters containing fewer genes, which had complex expression patterns involving changes in expression particular to the PBM (clusters `r getClusterName('inc', 'dec')` and `r getClusterName('dec', 'inc')`).

## Dynamic expression of transcription factor families

Examination of the clustered genes suggested that transcription factor (TF) genes were overrepresented in most clusters (Data S2). Using lists available in the Plant Transcription Factor Database [@PerezRodriguez:2010co], geneset enrichment analysis (GSEA) was used to provide an overview of expression of transcription factors and other regulators in the meristem samples (`r I(pasteLabel("Figure ", figCount, "f_gsea"))`). Unlike soft clustering, which was used to group genes based on shared expression dynamics, this analysis was used to investigate the overall strength of expression of the families based on Log~2~-fold changes (L~2~FC) in read count of their members. The most pronounced change in expression appears to occur between the RM and branch meristem samples (*i.e.* the change from apical to axillary meristem). Several families (including ABI3VP1 and GROWTH-REGULATING FACTOR TF genes and Aux/IAA genes) are more highly expressed in RM than in the other meristem types. Another group, including MYB and SBP TF genes and SET and PHD regulator genes, was enriched in one or both of the PBM and ePBM/AM samples but depleted or not enriched in the RM samples.

Homeodomain genes are involved at various stages of plant development and in several hormone response pathways [@Chan:1998wq; @Himmelbach:2002jr; @Sawa:2002vm]. The expression of this family in inflorescence meristem types was explored in greater detail using a heatmap of scaled, transformed read counts. Hierarchical clustering of the transformed counts recovered five prominent groups of expression (`r I(pasteLabel("Figure ", figCount, "f_hb"))`). In general, there was no clear relationship between homeodomain subfamily and expression pattern. However, there was an apparent enrichment of class IV HD-Zip genes with lowest expression in the RM, a peak in indeterminate axillary meristems and then a gradual decrease during axillary meristem transitions. 

# Discussion

Several previous transcriptomic studies have measured gene expression in whole rice inflorescences at later stages of development than those described here [*e.g.* @Wang:2010im; @Sato:2011eh; @Sharma:2012kq; @Khanday:2013bz; @Jiang:2014im]. Another study used microarray experiments at earlier stages to highlight the importance of transcription factors in panicle development [@Furutani:2006hx]. LMD has also been used for precise control of developmental stage during the collection of whole panicle sections for microarrays [@Kobayashi:2012fi]. This identified three MADS genes, which are members of the *SQUA*-like (*FUL*-like) clade, that are co-expressed with *PANICLE PHYTOMER2* (*PAP2*/*MADS34*), and a quadruple knockdown of these four genes resulted in defects in inflorescence development [@Kobayashi:2012fi]. All four genes were detected in all stages in the RNAseq dataset presented in this article (Data S1), confirming their expression in reproductive meristems. In contrast to previous transcriptomic datasets, only the meristematic regions of young panicles were collected by LMD for the RNAseq analysis described here, to restrict the measurement of gene expression to those tissues.

## Switch in gene expression between apical and axillary meristems

Clusters `r getClusterName('dec', 'small')` and `r getClusterName('inc', 'small')` contain genes that change in expression between apical (RM) and axillary meristems (PBM to SM). Axillary meristem initiation requires auxin synthesis and transport in *Arabidopsis thaliana* and maize [reviewed by @Gallavotti:2013ki], and excess auxin may be inactivated by conjugation to amino acids by GH3 enzymes [@Staswick:2005cv], but these processes are not as well understood in rice. Cluster `r getClusterName('inc', 'small')` contains the auxin response factor *ARF6A* and the small auxin-up RNA *SAUR33*, as well as two ABC transporters (*ALS1* and *MDR12*), while the complementary cluster `r getClusterName('dec', 'small')` contains four uncharacterized genes with annotations relating to auxin (*IAA10* and *IAA16*, two auxin-responsive transcriptional regulators; *ABCB27*, an ABC transporter; and *LOC_Os01g63770*, an AUX/LAX transporter), and the GH3 gene related to auxin homeostasis, *GH3-8*. Transgenic rice plants presumably expressing *GH3-8* in the endogenous pattern of *FLO-LFY HOMOLOG OF RICE* (*RFL*) [*i.e.* PBM and SBM; @IkedaKawakatsu:2012co] have smaller panicles with fewer branches [@Yadav:2011db], and GH3-8 conjugates amino acids to auxin *in vitro* [@Ding:2008hv], suggesting that the panicle phenotype may be related to reduced auxin signalling. Although auxin response factors and Aux/IAA response genes are generally more strongly expressed at earlier stages (`r I(pasteLabel("Figure ", figCount, "f_gsea"))`), the presence of auxin-related genes in complementary clusters suggests that the transcriptional effects of auxin signalling are dependent on meristem type. Understanding of the relationship between auxin and meristem specification will require functional analysis of the components of the regulatory network involved in the response in these tissues.

Two *STMADS11*-like (*SVP*-like) MADS genes, *MADS22* and *MADS55*, decrease in expression between PBM and SM, whilst the third (*MADS47*) was not detected (`r I(pasteLabel("Figure ", figCount, "f_alogFamily"))`a). *MADS22* and *MADS55* are activated by the *G1*-Like (G1L/ALOG) transcriptional activator *G1L5* (*TAWAWA1*/*TAW1*), but activation of *MADS47* has not been reported [@Yoshida:2013ff]. Overexpression of *TAW1* increases the production of secondary branches and leads to the formation of tertiary branches, resulting in higher grain yield [@Yoshida:2013ff]. *TAW1* was recovered from the same cluster as *MADS55*, and the presence and co-expression of genes that are involved in inflorescence development along with other TF genes indicates that the clusters capture biologically meaningful groups of genes, including uncharacterized signalling components. Only three ALOG genes (which are annotated with the Pfam domain PF04852) were detected in the dataset. Two were retrieved from clusters `r getClusterName('dec', 'small')` and `r getClusterName('dec', 'dec')` (*G1L2* and *TAW1* respectively), and the third (*G1L1*) has a similar expression pattern (`r I(pasteLabel("Figure ", figCount, "f_alogFamily"))`b). *taw1* missense mutants have small inflorescences with reduced grain yield, but this phenotype was stronger when *TAW1* was targeted by RNA interference [@Yoshida:2013ff]. This may suggest functional redundancy between the three co-expressed ALOG genes, *G1L1*, *G1L2* and *TAW1*, with more than one affected by the RNAi construct used against *TAW1*. Although the increase in secondary branches linked to *TAW1* overexpression suggests that it acts by suppression of SM identity, perhaps via downstream factors that may promote BM activity such as the *STMADS11*-like genes, *taw1* plants also have fewer primary branches [@Yoshida:2013ff], and the peak expression in the RM of *G1L1*, *G1L2*, *TAW1*, *MADS22* and *MADS55* suggests an additional role before the transition from apical to axillary meristem. The difference in expression of annotated transcription factor genes between RM and PBM samples and the recovery of co-expression clusters of primarily uncharacterized genes that appear to be either switched on or off between the RM and the PBM indicate a significant change in gene expression between apical and axillary meristems.

## Gradual changes in expression during transitions in axillary meristem identity

Compared to the switch from apical to axillary meristem, more gradual changes in gene expression were apparent during the sequential transition from PBM to SM (`r I(pasteLabel("Figure ", figCount, "f_mfuzzClusters"))`). This transition corresponds to the acquisition of determinate fate by the axillary meristem. The enrichment of MADS genes in the SM sample and their overrepresentation in cluster `r getClusterName('small', 'inc')` is consistent with their involvement in flower development in rice and other plants [`r I(pasteLabel("Figure ", figCount, "f_gsea"))`; *e.g.* @Kyozuka:2002vm; @Nagasawa:2003gu;  @Yamaguchi:2006br; @Yao:2008dy; @Dreni:2011bu; the role of MADS genes in Arabidopsis flower development was reviewed by @Prunet:2013ki]. Two recent studies in *A. thaliana* and the domesticated tomato, *Solanum lycopersicum*, used an LMD and RNAseq approach to investigate inflorescence development [@Park:2012dg; @Mantegazza:2014di]. These two datasets were used with the dataset presented in this article to compare the expression of the MIKC^C^-type class of MADS-box genes [@Henschel:2002ui], suggesting several differences between the species (`r I(pasteLabel("Figure S", s_figCount, "sf_madsTree"))`; Methods S1). *A. thaliana* *AGL6* is the only gene from the *AGL6*-like clade that is more highly expressed in indeterminate meristems than determinate meristems. *agl6* mutants have fewer axillary buds on the inflorescence stem, suggesting that *AGL6* promotes axillary meristem formation [@Huang:2012de]. The rice genes from the *AGL6*-like clade, *MFO* (*MADS6*) and *MADS17*, redundantly promote floral meristem determinancy [@Ohmori:2009kj], and are correspondingly enriched in determinate meristems. Expression of the putative tomato *AGL6*-like gene *SOLYC01G093960.2* is also enriched in determinate meristems, suggesting that the role of *AGL6* in axillary meristem formation may be specific to the Arabidopsis lineage. In contrast, consistent with their functions in floral organ specification in rice and *A. thaliana*, genes in the *AGL2*-like clade are generally more highly expressed in determinate meristems. *PAP2*, which was strongly expressed at all stages with higher expression in the RM and PBM (Data S1), acts with another *AGAMOUS-LIKE2*-like (*AGL2*-like/*SEPELLATA*) MADS gene, *LEAFY HULL STERILE 1* (*LHS1*/*MADS1*), to specify floral organs [@Gao:2010iz]. *pap2* mutants also have altered primary and secondary branch number, suggesting an earlier role is spikelet meristem specification [@Gao:2010iz; @Kobayashi:2010hn]. *LHS1*, which is involved in floral meristem determination and floral organ development [@Jeon:2000tj; @Agrawal:2005ho], was enriched in the SM (`r I(pasteLabel("Figure S", s_figCount, "sf_madsTree"))`).

The strong enrichment of SQUAMOSA promoter binding protein-like (SPL/SBP) genes in the PBM and their presence in cluster `r getClusterName('small', 'dec')` suggests involvement in the early stages of inflorescence development. *WEALTHY FARMER'S PANICLE* (*WFP*/*SPL14* or *IDEAL PLANT ARCHITECTURE 1*/*IPA1*), *SPL3*, *SPL12* and *SPL17* are all highly expressed in the RM and PBM, gradually decrease in expression and have weakest expression in the SM (`r I(pasteLabel("Figure ", figCount, "f_alogFamily"))`c), suggesting a redundant function in indeterminate axillary meristems. These genes are all putative targets of *miR156*, a microRNA that causes reduced panicle size and delayed flowering when overexpressed [@Xie:2006dc; @Wang:2015jx], and *SPL* gene expression is controlled by *miR156* and *miR529* [@Jeong:2011gv; @Wang:2015jx]. The precursor for another microRNA, *MIR319A*, was recovered from the same cluster as *SPL3* and *SPL12* and has a similar expression pattern to these genes, *SPL17* and *WFP* (`r I(pasteLabel("Figure ", figCount, "f_alogFamily"))`c). In *A. thaliana*, *miR319a* targets *TEOSINTE BRANCHED*/*CYCLOIDEA*/*PROLIFERATING CELL FACTORS* (*TCP*) transcription factors and is involved in floral organ development [@Palatnik:2003kt; @Nag:2009fc]. No direct interaction with *SPL* genes or role in reproductive development has been reported for *miR319a* in rice. Overexpression of *WFP*, which was detected by *in situ* hybridization experiments in the RM and branch meristems in wild-type plants, is associated with increased branching and grain yield [@Jiao:2010ft; @Miura:2010it]. WFP binds to the promoter region and appears to cause upregulation of the *DENSE AND ERECT PANICLE 1* (*DEP1*) gene [@Lu:2013hy], which is present in cluster `r getClusterName('inc', 'dec')` and encodes a major QTL for grain production and panicle morphology in high-yield *O. sativa* ssp. *japonica* varieties. Probable loss-of-function *dep1* mutants have larger SAMs with higher cell number, and reduced expression of a cytokinin oxidase gene, *GRAIN NUMBER 1A* (*GN1A*/*CKX2*) [@Huang:2009kf]. WFP also binds to the promoters of genes that are involved in the determination of inflorescence architecture via the cytokinin, auxin and gibberelic acid pathways, including *LONELY GUY 1* (*LOG1*), *PIN PROTEIN 1B* and *SLENDER RICE1* respectively, suggesting a role in the coordination of hormone signalling [@Lu:2013hy].

A homolog of *LOG1*, *LONELY GUY LIKE PHOSPHORIBOHYDROLASE 1* (*LOGL1*), is present in cluster `r getClusterName('inc', 'inc')` with genes that increase in expression between RM and SM. In rice, cytokinins may increase branching complexity by promoting IM or BM activity [reviewed by @Han:2014dt]. The LOG1 enzyme converts cytokinin nucleotides to the active form *in vitro*, and *log1* mutants have lower expression of two cytokinin-inducible *RESPONSE REGULATOR* genes, resulting in early termination of BM and IM and small panicles with small SAMs and branching defects [@Kurakawa:2007go]. Similarly, reduced expression of *GN1A*, which encodes an enzyme that inactivates cytokinin, is associated with increased grain production in several high-yield *O. sativa* ssp. *indica* varieties [@Ashikari:2005eg]. Although *GN1A* transcripts were not detected in this LMD dataset (`r I(pasteLabel("Figure ", figCount, "f_alogFamily"))`d), these results suggest that an increase in active cytokinin is associated with grain yield via an effect on meristem activity. However, the function of the *LONG AND BARBED AWN1* (*LABA1*/*LOGL6*/*An-2*) gene provides evidence for an inverse role. The LABA1 protein present in *O. rufipogon*, which shares an identical primary sequence with the protein from *O. sativa* ssp. *japonica* cv. Nipponbare, also activates cytokinin *in vitro* [@Gu:2015hy; @Hua:2015bs]. Introduction of the *Laba1* allele from *O. rufipogon* into an *O. sativa* ssp *indica* background, which contains a non-functional *laba1* allele, results in a higher concentration of endogenous cytokinins and higher expression of *RESPONSE REGULATOR* genes [@Hua:2015bs], but reduces the number of grains per panicle and the number of tillers per plant [@Gu:2015hy]. *laba1* is a domestication allele found at high frequencies in cultivated accessions, implying that it was affected by artificial selection either for awn phenotype or grain yield [@Gu:2015hy; @Hua:2015bs]. *LABA1* was detected in the ePBM/AM and SM samples but not the RM or PBM samples. Although the function of LOGL1 has not been reported, the expression patterns of *LABA1* and *LOGL1* suggest increasing cytokinin activation along the course of axillary meristem determination (`r I(pasteLabel("Figure ", figCount, "f_alogFamily"))`d). Such a mechanism would support a cytokinin function secondary to the maintenance of IM and BM, but it is not clear how this would influence meristem identity or panicle branching at the molecular level. A type-A *RESPONSE REGULATOR* (*RR3*) is present in cluster `r getClusterName('dec', 'small')` (containing genes more highly expressed in RM), highlighting the possible complexity of the roles of cytokinins in inflorescence development.

## Homeodomain genes and meristem identity

*log1* mutants also appear to have reduced *HOMEOBOX 1* (*OSH1*) expression [@Kurakawa:2007go]. *OSH1* encodes a class I *Knotted1-like homeobox* (*KNOX*) homeodomain transcription factor, and the expression of class I *KNOX* genes in meristems in the developing inflorescence has been detected by *in situ* hybridisation [@Sentoku:1999ti]. *osh1* mutants have defects in SAM maintenance during vegetative growth and lower expression of other *KNOX* genes, and expression of *KNOX* genes can be induced by cytokinin treatment during shoot regeneration [@Tsuda:2011gz]. Another class I *KNOX* gene, *HOMEOBOX 15* (*OSH15*) is expressed in a similar pattern to *OSH1* (`r I(pasteLabel("Figure ", figCount, "f_hb"))`), as previously reported in *O. sativa* cv. Nipponbare [@Sato:1998tp], and a third, *OSH6*, has a peak in expression in the RM and PBM (`r I(pasteLabel("Figure ", figCount, "f_hb"))`). OSH1 binds to the upstream region of the *OSH15* locus, and *osh1 osh15* double-mutants have lower induction of *OSH6* after treatment with cytokinin [@Tsuda:2011gz], supporting a role of *KNOX* genes in the response to this phytohormone. The higher expression of *OSH1* and *OSH15* in indeterminate axillary meristems is compatible with a function in the promotion of BM identity. In *A. thaliana*, cytokinin promotes the expression of the homeodomain gene *WUSCHEL*, which itself enhances cytokinin signalling [@Leibfried:2005hw; @Lindsay:2006bh; @Gordon:2009fp]. *wox1* plants carrying a likely null mutation in the rice ortholog of *WUSCHEL*, *WUSCHEL-LIKE HOMEOBOX 1* (*WOX1*/*WUS*) [@Nardmann:2006fd], have defects in axillary meristem formation and lower expression of *OSH1* [@Tanaka:2015ko]. *WOX1* was not detected in the meristem tissues described here, but another *WUSCHEL homeobox* (*WOX*) gene, *WOX8*, has a peak in expression in RM, and a third, *LOC_Os01g70810*, is predominantly expressed in the RM and PBM (`r I(pasteLabel("Figure ", figCount, "f_hb"))`). Although these results and previous studies support a role of *KNOX* and *WOX* genes in the maintenance of indeterminate meristem identity, possibly coordinated by phytohormone signalling, more work is required to elucidate this mechanism.

In *A. thaliana*, class III homeodomain-leucine-zipper (HD-Zip) genes are involved in meristem initiation and regulation [@Prigge:2005gj]. The class III HD-Zip gene *REVOLUTA* is required for axillary meristem formation [@Talbert:1995wa; @Otsuga:2001us]. Seven of the nine annotated class III HD-Zip genes in *O. sativa* were detected (`r I(pasteLabel("Figure ", figCount, "f_hb"))`), including two homologs of *REVOLUTA*, *HOX9* and *HOX10* [@Prigge:2006ky], which are both highly expressed in the RM. Three class IV HD-Zip genes, *RICE OUTMOST CELL-SPECIFIC GENE 1* (*ROC1*), *ROC3* and *LOC_Os09g35760*, were recovered in cluster `r getClusterName('inc', 'small')`, containing genes that had lowest expression in the RM and higher expression in the samples from axillary meristems. Five of the eight detected class IV HD-Zip genes and several genes from other homeobox subclasses, including *OSH1*, also follow this pattern (`r I(pasteLabel("Figure ", figCount, "f_hb"))`). *Zea mays* class IV HD-Zip genes are expressed in the outer cell layer of the SAM and transgenic overexpression of *OUTER CELL LAYER 1* delays flowering [@Javelle:2011dz; @DepegeFargeix:2011ds]. In rice, several class IV HD-Zip genes have been detected in the vegetative SAM by *in situ* hybridisation [@Ito:2003ta], but their role in axillary meristems in the developing inflorescence has not been explained. There are also smaller groups of homeobox genes that peak in expression in the SM or RM samples, suggesting multiple roles and possible redundancy for homeodomain proteins during inflorescence development.

## Conclusion

The molecular mechanisms controlling the meristematic activities that guide panicle development in rice remain largely uncharacterized. The dataset presented here addresses this by describing both the transcriptome profiles associated with specific meristem identities and the changes in gene expression that occur during the early stages of panicle development. A marked expression switch was evident between apical and axillary meristems, followed by gradual changes during transitions between PBM and SM. This could be explained by the difference between the fates of the rachis meristem, which aborts after PBM differentiation [@Ikeda:2004tk], and the axillary meristems, which undergo transition to determinate SMs. The latter process may be comparable to the gradual meristem maturation observed during tomato inflorescence development [@Park:2012dg], rather than to the continual production of determinate floral meristems characteristic of the Arabidopsis inflorescence meristem. Promising uncharacterized genes were identified, notable examples including transcription factor genes with expression patterns similar to known regulators of panicle architecture, and putative hormone-related genes such as the auxin-responsive genes. Functional studies of these genes will reveal the mechanics of the regulatory networks that establish gene expression patterns in meristems and their control by hormone signalling. Genes with expression specific to certain meristem types were also identified, which may prove valuable as marker genes, for example in developmental studies of mutants affected in determination of reproductive meristem identity. Combined with current technology for the creation of mutants such as CRISPR/Cas9 genome editing, this dataset promises to accelerate research on the molecular mechanisms controlling panicle development, which is a vital for efforts to achieve the sustainable increase in grain yield required to address population growth.

# Experimental procedures

## Plant material and sampling

*O. sativa* ssp. *japonica* cv. Nipponbare plants were grown in a growth chamber with 70% relative humidity at 30°C during the day and 26°C at night. After 7 weeks in long day conditions (14 hours light and 10 hours darkness) plants were transferred to short day conditions (10 hours light and 14 hours darkness) to induce floral transition. Inflorescence meristems were harvested from 6 to 14 days after the change of photoperiod.

## Morphological analysis

Panicles were harvested in FAA (formaldehyde 10%, acetic acid 5%, ethanol 50%), infiltrated under vacuum for 15 minutes, stored overnight at 4°C and embedded in paraffin as described by @Huijser:1992wz. Tissues were cut into 8 µm sections using an RM2155 microtome (Leica), mounted on glass slides and stained with 0.5% w/v toluidine blue for observation with an Axiophot D1 microscope (Zeiss) and image capture with an Axiocam MRc 5 camera (Zeiss).

## Tissue embedding, laser microdissection and sequencing

Each inflorescence was harvested in 2 mL ice-cold 3:1 ethanol:acetic acid fixative solution, infiltrated twice under a mild vacuum for 15 minutes and stored for 20 hours in fresh fixative at 4°C. Embedding for dissection was performed as described by @Mantegazza:2014di. Tissues were cut into 8 µm sections on an RM2155 or RM2255 microtome (Leica) and dissected on a LMD6000 or LMD7000 laser dissector (Leica). RNA was amplified using the ARCTURUS PicoPure RNA Isolation Kit (ThermoFisher) and assayed on a 2100 Bioanalyzer with the RNA 6000 Pico Kit (Agilent) before amplification and cDNA synthesis with the Ovation RNA-Seq System V2 (NuGEN). Library preparation with the Ovation Ultralow Library Sytem (NuGEN) and 50-base, single-end sequencing on the HiSeq 2000 platform (Illumina) were performed by IGA Technology Services (Udine, Italy).

## RNA *in situ* hybridization

Samples harvested from the main stem from four developmental stages of panicle development were embedded in Paraplast X-TRA (Sigma-Aldrich) as described by @Huijser:1992wz. Digoxigenin-labelled antisense and sense RNA probes were generated with the DIG RNA Labeling Kit SP6/T7 (Roche) according to the manufacturer's instructions. To generate the probes, cDNA was amplified using the following primers: *LOC_Os01g04670*, 5′-GTGTCAAGGCATCGCCAAC-3′ and 5′-CATCAGCTGGCTGCTTTACC-3′; *LOC_Os10g04270*, 5′-TCCCAGTTGACCGAGAACTG-3′ and 5′-TGAACTTCCGTCACGAACTCC-3′; *LOC_Os10g05990*, 5′-CCTCCGGCAAAGAACTGATG-3′ and 5′-CCGTCATTGGGACTAGTTTGTCAG-3′. For detection of *LOC_Os09g27730*, a Digoxigenin-labelled LNA probe with the sequence 5′-TCTG{A}CGACG{T}GCG{A}C{T}GGT-3′ was synthesised (Eurogentec, Liège, Belgium). Hybridization was performed as described by @Coen:1990ux using NBT/BCIP (Roche) or VECTOR Blue Alkaline Phosphatase Substrate Kit (Vector Laboratories) for detection.

## Data analyses

Analysis of the sequenced libraries was carried out with `bash` and `R` scripts [@RALanguageandEn:2015wf], which were arranged into a pipeline using the `ruffus` package for `python3` [@Goodstadt:2010jp]. All of the code used for the analysis and to generate text, figures and tables for this report, along with the versions and parameters of the software used, are available with revision history in a public GitHub repository at [https://github.com/evoreprice/lmdPaper](https://github.com/evoreprice/lmdPaper).

Briefly, adaptor trimming was performed with `cutadapt` [@Martin:2011eu] and reads were mapped against the MSUv7 genome and annotation [@Kawahara:2013gh; @Ouyang:2007il] downloaded from Phytozome 10.3 [@Goodstein:2012ho] using `STAR` [@Dobin:2012fg] in 2-pass mode. The `'quantMode'` argument of `STAR` was used for read counting. rRNA and tRNA contamination was estimated using `htseq-count` [@Anders:2014gf] to count the number of reads that mapped to regions annotated as tRNA or rRNA on the Rap-DB [@Ohyanagi:2006go] or to regions where reads simulated with `wgsim` ([https://github.com/lh3/wgsim](https://github.com/lh3/wgsim)) from rRNA sequences in the TIGR Plant Repeats Database [@Ouyang:2004ia] also mapped. Differential expression analysis to calculate Log~2~-fold change (L~2~FC) values and transformed read counts was performed with the `DESeq2` package [@Love:2014do]. To determine a strict cut-off for unexpressed genes, expression values in transcripts per million were calculated for each gene [@Li:2010cl; @Wagner:2012jf]. These values were compared to pseudo-expression values calculated from intergenic regions of the genome, using intervals of a similar size distribution to those used for calculating the 'genic' expression values. For each library, the cut-off was placed at the 95^th^ percentile of the distribution of intergenic expression values. Transformed counts and L~2~FC values from expressed genes were used for soft clustering with the `Mfuzz` package [@Kumar:2007uw] and geneset enrichment analysis with the `gage` package [@Luo:2009kp] respectively. Gene lists were analysed using annotations from several databases, including TIGR [@Kawahara:2013gh], Oryzabase [@Kurata:2006gb] and OGRO [@Yamamoto:2012kw]. All plots were produced with the `ggplot2` package [@Wickham:2009tn].

# Accession Numbers

All rice genes are referred to by either their locus identifier [MSUv7; @Ouyang:2007il] or their official gene symbol provided on Oryzabase [@McCouch:2008ef].

# Acknowledgements

We thank James Tregear for feedback and discussion, and Jean-Luc Verdeil and Geneviève Conejero for assistance with laser microdissection at the PHIV imaging platform (CIRAD, Monpellier, France). This project was jointly supported by Fondation Agropolis through the *Investissements d'avenir* programme (ANR-10-LABX-0001-01) and Fondazione Cariplo under the EVOREPRICE 1201-004 project.

# Supporting Information

**`r I(pasteLabel("Figure S", s_figCount, "sf_pca"))`**: Principal components analysis of transformed read counts for each library.

**`r I(pasteLabel("Figure S", s_figCount, "sf_isGenesTpm"))`**: Comparison of the LMD RNA-sequencing dataset with published *in situ* results.

**`r I(pasteLabel("Figure S", s_figCount, "sf_mfuzzPcaCentroids"))`**: Assessment of clustering results.

**`r I(pasteLabel("Figure S", s_figCount, "sf_madsTree"))`**: Differential expression
of genes encoding putative MIKC proteins in *A. thaliana*, *S. lycopersicum* and *O. sativa* ssp. *japonica*.

**`r I(pasteLabel("Table S", s_tableCount, "st_libStats"))`**: Library and sequencing statistics.

**`r I(pasteLabel("Table S", s_tableCount, "st_refTable"))`**: References for `r I(pasteLabel("Figure S", s_figCount, "sf_isGenesTpm"))`.

**Data S1**: Expression values for all genes in transcripts per million with annotations.

**Data S2**: List of clustered genes with annotations.

# Figure legends

![](data/lmdFigure.pdf)

**`r I(pasteLabel("Figure ", figCount, "f_lmdFigure"))`**. Morphology of the early stages of inflorescence development and laser microdissection of meristem samples. Toluidine blue stained sections of developing panicles at rachis meristem (RM; **a**), primary branch meristem (PBM; **b**), elongating primary branch meristem with axillary merisem (ePBM/AM; **c**) and spikelet meristem (SM; **d**) stages of differentiation. The position of the first bract primordium (bp) and flag leaf (fl) are indicated in **a**. Laser microdissection (LMD) samples were collected from RM (**e** and **i**), PBM (**f** and **j**), ePBM/AM (**g** and **k**) and SM (**h** and **l**). Images show the samples before (**e**–**f**) and after (**i**–**l**) dissection. Scale bars represent 50 µm (**a**–**c**), 100 µm (**d**), 200 µm (**e**–**g** and **i**–**k**) or 320 µm (**h** and **l**).

----

![](data/isFigure.pdf)

**`r I(pasteLabel("Figure ", figCount, "f_inSitu"))`**. RNA *in situ* hybridization analysis to confirm the specific expression patterns of four genes detected in the RNA sequencing dataset. Expression of *LOC_Os09g27730* (**a**–**d**), *LOC_Os01g04670* (**e**–**h**), *LOC_Os10g04270* (**i**–**l**) and *LOC_Os10g059908* (**m**–**p**) was analysed at the RM (**a**, **e**, **i** and **m**), PBM (**b**, **f**, **j** and **n**), ePBM/AM (**c**, **g**, **k** and **o**) and SM (**d**, **h**, **l** and **p**) stages. Each gene was detected in the same stage by *in situ* hybridization as by RNA sequencing. The scale bars represent 100 µm.

____

```{r f_mfuzzClusters, fig.height = 8.661 / 2}
print(f_mfuzzClusters)
```

**`r I(pasteLabel("Figure ", figCount, "f_mfuzzClusters"))`**. Common patterns of gene expression in developing inflorescences. Fuzzy *c*-means clustering of normalised, variance-stabilised read counts was used to recover `r as.character(english::english(c))` common patterns of expression. Each line describes the expression pattern of one gene, with the gene's membership to the cluster represented by the colour of the line. The core values for each cluster are plotted in black. The ordering of the panels in the plot was arranged to enable side-by-side comparison of complementary clusters.

----

```{r f_gsea, fig.height = 8.661 / 2}
print(f_gsea)
```

**`r I(pasteLabel("Figure ", figCount, "f_gsea"))`**. Geneset enrichment analysis (GSEA) of transcription factor and other transcriptional regulator families. Test statistics were calculated from the Log~2~-fold change between the stage of interest and all other stages for the expressed genes in each geneset. The absolute value of the test statistic indicates the magnitude of the geneset-level change, and a positive test statistic indicates enrichment and a negative statistic indicates depletion.

----

```{r f_hb, fig.height = 8.661}
print(f_hb)
```

**`r I(pasteLabel("Figure ", figCount, "f_hb"))`**. Five prominent patterns of expression of homeobox genes in reproductive meristems. Scaled, transformed read counts for the homeobox genes that were detected in the dataset are represented by the continuous heat scale. The colour of the *y*-axis denotes the homeobox subfamily for each gene [@Jain:2008kg], and the number of genes detected and the total number of genes for each subfamily is given in brackets after the subfamily name in the legend. The expression patterns were separated into five groups using hierarchical clustering, including a set of homeobox genes with lowest expression in the RM and highest expression in indeterminate axillary meristems, which contains *OSH1* and five of the eight expressed class IV HD-Zip genes.

----

```{r f_alogFamily, fig.width = 6.614, fig.height = 6.614}
# set layout
grid::grid.newpage()
grid::pushViewport(grid::viewport(layout = grid::grid.layout(4, 4)))

# print the plots
print(f_alogFamily_a, vp = grid::viewport(layout.pos.row = 1:3, layout.pos.col = 1))
print(f_alogFamily_b, vp = grid::viewport(layout.pos.row = 1:3, layout.pos.col = 2))
print(f_alogFamily_c, vp = grid::viewport(layout.pos.row = 1:3, layout.pos.col = 3:4))
print(f_alogFamily_d, vp = grid::viewport(layout.pos.row = 4, layout.pos.col = 1:4))
```

**`r I(pasteLabel("Figure ", figCount, "f_alogFamily"))`**. Expression of selected genes  in transcripts per million (TPM). A red point indicates that the gene was above the detection cutoff, and blue indicates that the gene was not detected. (**a**) Two reported *STMADS11*-like (*SVP*-like) targets of TAW1, *MADS22* and *MADS55* [@Yoshida:2013ff], are more strongly expressed in the RM than in other meristems, and the other *STMADS11*-like gene, *MADS47*, was only detected in one library. (**b**) Only three ALOG genes (*G1L1*, *G1L2* and *G1L5*/*TAW1*) were detected, and they share a similar pattern of expression. (**c**) Several *SPL* genes and a co-regulated microRNA precursor, *MIR319A*, which was recovered in cluster `r getClusterName('small', 'dec')`, are highly expressed in RM and PBM before decreasing in expression in ePBM/AM and SM. (**d**) *GN1A* (*CKX2*) was not detected in the LMD dataset, but *LOG1* is expressed in all meristem types, and two other genes possibly related to cytokinin activation, *LABA1* (*LOGL6*) and *LOGL1*, are expressed more highly in ePBM/AM and SM than in RM and PBM. 

# References (move to before **Figure Legends**)
 