---
title: Genome-wide changes in expression during inflorescence architecture development
  in rice
csl: csl/the-plant-journal.csl
header-includes: \usepackage{graphicx}
output:
  BiocStyle::html_document:
    fig_caption: yes
    force_captions: yes
    highlight: pygments
    keep_md: yes
    number_sections: yes
    dev: "CairoPDF"
  word_document:
    fig_caption: yes
  pdf_document:
    fig_caption: yes
    keep_tex: yes
    number_sections: yes
bibliography: bib/test.bib
---

```{r Setup, include=FALSE, results="hide", warning=FALSE}

knitr::opts_chunk$set(fig.path='figures/',
               cache.path='cache/',
               dev=c("svg", "cairo_pdf"),
               fig.width=3.150,
               fig.height=6.3,
               fig.align = 'center',
               dpi=300,
               fig.show='hold',
               cache=FALSE,
               echo=FALSE,
               results="hide",
               message=FALSE,
               warning=FALSE)

source('src/R/figures.R')

```

Authors:

Addresses:

### Running title

### Keywords

### SRA details for libaries

Word count

***

# Summary

# Significance statement

# Introduction

* Rice inflorescence development
+ Branching & grain yield
+ Sequential changes in meristem activity
+ Discuss a few key genes

# Results and Discussion

To investigate gene expression during inflorescence development, LASER microdissection (LMD) was used to collect meristematic tissues from the rachis meristem (RM), primary branch meristem (PBM), elongating primary branch meristem and secondary branch meristem (ePBM/SBM) and spikelet meristem (SM) (`r I(pasteLabel("Figure ", figCount, "f_lmdFigure"))`). RNA from the meristem samples was isolated, amplified and used for the production of a sequencing library. **need area cut numbers**. RNA input for library preparation was between `r min(st_libStats$"Yield (ng)")` and `r max(st_libStats$"Yield (ng)")` ng and the average RNA integrity number was `r round(mean(st_libStats$"RIN"), 1)` (`r I(pasteLabel("Table S", s_tableCount, "st_libStats"))`). Initially, two biological replicates were prepared for each stage, but after performing a principal components analysis (PCA) on transformed expression values two libraries that were made from degraded RNA were excluded from the analysis (`r I(pasteLabel("Figure S", s_figCount, "sf_pca"))`). Following this, further samples were prepared to provide a total of three biological replicates with intact RNA (RIN ≥ 7) for each meristem type.

![](data/lmdFigure.pdf)

At least `r round(min(st_libStats$"Reads (M)"), 0)` million 50-base reads were produced from each library, yielding between `r round(min(st_libStats$"Reads in genes (M)", na.rm = TRUE), 0)` and `r round(max(st_libStats$"Reads in genes (M)", na.rm = TRUE), 0)` million reads that mapped to genes after removal of reads likely to have originated from rRNA and tRNA. Using cutoffs calculated from the 95^th^ percentile of distributions of expression values estimated from intergenic regions for each library (*i.e*. reads that were not generated from genes), `r length(unique(readRDS("output/expressedGenes/expressedGenesAll.Rds")))` unique genes were detected in at least one of the libraries.

##  Patterns of gene expression during inflorescence development

To recover common expression patterns in genes detected in the meristem samples, read counts were transformed using the variance-stabilizing transformation (VST) included in the `DESeq2` package [@Love:2014do]. Expressed genes were ranked by variance, and standardised, VST-transformed read counts for the 25% most-variable genes were clustered using the fuzzy *c*-means algorithm implemented in the `Mfuzz` package [@Futschik:2005vo]. To determine the number of cluster cores (*c*), the minimum distance between cluster centroids, the formation of empty clusters, and PCA plots of cluster members were monitored for clusters produced with *c* values between 2 and `r maxClust` (`r I(pasteLabel("Figures S", s_figCount, "sf_mfuzzCentroids"))` `r I(pasteLabel(" and S", s_figCount, "sf_mfuzzPca"))`), leading to the use of `r as.character(english::english(c))` cluster cores (`r I(pasteLabel("Figure ", figCount, "f_mfuzzClusters"))`). Clusters were recovered that contain genes that gradually decrease or increase in expression during development (clusters **2**, **3**, **6** and **8**), or that are apparently turned on or off in primary and secondary branch meristems (clusters **7** and **4** respectively) along with genes with more complex expression patterns (clusters **1** and **5**).

```{r f_mfuzzClusters, fig.height=6.3}
print(f_mfuzzClusters)
```

Examination of the clustered genes suggested that transcription factor genes were overrepresented in most clusters (**annotated cluster tables in SI**). To determine whether particular TF families were associated with particular patterns of expression, the enrichment of TF family members in each cluster was calculated using hypergeometric tests. Using the list of transcription factor families provided by the Plant TFDB (**reference**), a hypergeometric test was carried out if there were at least two member of TF family in a given cluster. After adjustment for multiple comparisons, each cluster had a significant enrichment of at least one TF family (`r I(pasteLabel("Table ", tableCount, "t_hypergeom"))`). For example, members of the ABI3VP1 family were overrepresented in clusters **2** and **4**, both of which contain genes with higher transformed read counts in the rachis meristem than in samples from later stages of development. The MADS family was overrepresented in clusters **6** and **8**, suggesting that they could be involved in both early and late stages of inflorescence development.

+ Clusters including characterised genes

## Patterns of expression of transcription families

+ **Figure**: GSEA of transcription factor families

Because of the enrichment of TF family members in some of the gene expression clusters, geneset-level enrichment analysis (GSEA) was used to compare the levels of expression of TF family members in the meristem samples. **Something like: Consistent with their enrichment in cluster 1 and 6, MADS genes were significantly enriched in the SM/FM samples**. TF families involved in branching were enriched in **which families, which samples**.

## MADS genes: comparison of expression & homology in rice, tomato, arabidopsis

The presence of MADS genes was enriched in **some clusters** and their overall expression is enriched in the spikelet meristem sample. These results correspond with previous studies that have established the importance of MADS genes in inflorescence development. The evolution of their expression patterns was 

+ **Figure**: Similarity tree of MADS proteins with expression values and dN/dS values for each one.

# Experimental procedures

# Accession Numbers

# Acknowledgements

# Short legends for Supporting Information

**`r I(pasteLabel("Figure S", s_figCount, "sf_pca"))`**: Principal components analysis of transformed read counts for each library.

**`r I(pasteLabel("Figure S", s_figCount, "sf_mfuzzCentroids"))`**: Minimum centroid distance for 2–`r maxClust` cluster cores (*c*).

**`r I(pasteLabel("Figure S", s_figCount, "sf_mfuzzPca"))`**: Classical multidimensional scaling of Euclidean distances between transformed read counts for clustered genes.

**`r I(pasteLabel("Table S", s_tableCount, "st_libStats"))`**: Library and sequencing statistics.

# References

***

# Tables

```{r hypergeom, results = "markup"}
knitr::kable(t_hypergeom, caption = paste0(
  I(pasteLabel("**Table ", tableCount, "t_hypergeom")),
  '**: Transcription factor family enrichment per cluster'))
```

***

# Figure legends

**`r I(pasteLabel("Figure ", figCount, "f_lmdFigure"))`**. LASER microdissection of meristem samples in the early stages of inflorescences development. Samples were collected from the rachis meristem (**a** and **e**), the primary branch meristem (**b** and **f**), the elongating primary branch meristem and secondary branch meristem (**c** and **g**) and the spikelet meristem (**d** and **h**). Images show the samples before (**a**–**d**) and after (**e**–**h**) dissection.

**`r I(pasteLabel("Figure ", figCount, "f_mfuzzClusters"))`**. Common patterns of expression in the meristem samples. Fuzzy *c*-means clustering of normalised, variance-stabilised read counts was used to recover `r as.character(english::english(c))` common patterns of expression. Each line describes the expression pattern of one gene, with the gene's membership to the cluster represented by the colour of the line. The core values for each cluster are plotted in black.