---
title: Genome-wide changes in expression during inflorescence architecture development
  in rice
csl: csl/the-plant-journal.csl
header-includes: \usepackage{graphicx}
output:
  word_document:
    fig_caption: yes
  pdf_document:
    fig_caption: yes
    keep_tex: yes
    number_sections: yes
  BiocStyle::html_document:
    fig_caption: yes
    force_captions: yes
    highlight: pygments
    keep_md: yes
    number_sections: yes
bibliography: bib/test.bib
---

```{r Setup, include=FALSE, results="hide", warning=FALSE}

library(data.table)
library(ggplot2)
extrafont::loadfonts()

knitr::opts_chunk$set(fig.path='figures/',
               cache.path='cache/',
               dev=c("cairo_pdf"),
               fig.width=6.614,
               fig.height=6.614,
               fig.align = 'center',
               dpi=300,
               fig.show='hold',
               cache=TRUE,
               echo=FALSE,
               results="hide",
               message=FALSE,
               warning=FALSE)

# https://gist.githubusercontent.com/rmflight/3858973/raw/591ee603ce0996ac407902f7dd59520b35b3702a/tableFigureFuncs.r
pasteLabel <- function(preText, inObj, objName, insLink=TRUE){
	objNum <- inObj[objName]
	
	useText <- paste(preText, objNum, sep=" ")
	if (insLink){
		useText <- paste("[", useText, "](#", objName, ")", sep="")
	}
	useText
}

# this increments the counter, and gives a name to the number so we can reference it later
incCount <- function(inObj, useName){
	nObj <- length(inObj)
	useNum <- max(inObj) + 1
	inObj <- c(inObj, useNum)
	names(inObj)[nObj+1] <- useName
	inObj
}
figCount <- c("_"=0)
tableCount <- c("_"=0)

tableCat <- function(inFrame){
	outText <- paste(names(inFrame), collapse=" | ")
	outText <- c(outText, paste(rep("---", ncol(inFrame)), collapse=" | "))
	invisible(apply(inFrame, 1, function(inRow){
		outText <<- c(outText, paste(inRow, collapse=" | "))
	}))
	return(outText)
}

capwords <- function(s, strict = FALSE) {
    cap <- function(s) paste(toupper(substring(s, 1, 1)),
                  {s <- substring(s, 2); if(strict) tolower(s) else s},
                             sep = "", collapse = " " )
    sapply(strsplit(s, split = " "), cap, USE.NAMES = !is.null(names(s)))
}


```

Authors:

Addresses:

### Running title

### Keywords

### SRA details for libaries

Word count

***

# Summary

# Significance statement

# Introduction

* Rice inflorescence development
+ branching & grain yield
+ sequential changes in meristem activity
+ few key genes

# Results

## Dissections in early stages of inflorescence development

+  **Figure**: laser microdissections

## Transcriptome sequencing of meristem samples

+ Max. 1 paragraph about boring sequencing stats
+ **Table** (or supplementary): # reads, mapping %, rRNA, introns, genes blah blah

## Clustering of expression patterns

```{r mfuzz}
expressionMatrix <- readRDS('output/cutadapt-2015-07-29/STAR-2015-07-29/DESeq2-2015-08-04/mfuzz-2015-08-05/expressionMatrix.Rds')
c1 <- readRDS('output/cutadapt-2015-07-29/STAR-2015-07-29/DESeq2-2015-08-04/mfuzz-2015-08-05/c1.Rds')
memCutoff <- 0.5

# get clusters and membership
cluster <- data.table(id = names(c1$cluster), Cluster = c1$cluster, Membership = apply(c1$membership, 1, max), key = "id")

# get standardised VST counts
exprs <- data.table(Biobase::exprs(expressionMatrix), keep.rownames = TRUE, key = "rn")

# set up cluster labels
numGenes <- cluster[Membership > 0.5, .(number = length(id)), by = Cluster]
setkey(numGenes, 'Cluster')
clustLabels <- numGenes[, paste0("Cluster ", Cluster, "\n(", number, " genes)")]

# merge clusters with expression values
plotData.wide <- cluster[exprs]

# make long
plotData <- reshape2::melt(plotData.wide[ Membership > memCutoff ], id.vars = c('id', 'Cluster', 'Membership'), variable.name = 'Stage', value.name = "Normalised, transformed read counts")

# add cluster labels
plotData[, Cluster := plyr::mapvalues(as.factor(Cluster), seq(1, length(unique(Cluster))), clustLabels)]

# fix stage label
plotData[, Stage := plyr::mapvalues(Stage, "ePBM.SBM", "ePBM/SBM")]

# set up heatscale
heatscale <- RColorBrewer::brewer.pal(n = 6, name = "YlOrRd")

# add centres
centres.wide <- data.table(c1$centers, Cluster = clustLabels)
centres <- reshape2::melt(centres.wide, id.vars = 'Cluster', variable.name = 'Stage', value.name = "Normalised, transformed read counts")
centres[, Stage := plyr::mapvalues(Stage, "ePBM.SBM", "ePBM/SBM")]

# number of clusters (for text)
c <- length(unique(c1$cluster))

f_mfuzzClusters <- ggplot(plotData, aes(x = Stage, y = `Normalised, transformed read counts`,
                     colour = Membership, group = id)) +
  theme_minimal(base_size = 8, base_family = "Helvetica") +
  xlab(NULL) +
  scale_colour_gradientn(colours = heatscale, limits = c(0, 1), breaks = seq(0, 1, 0.2)) +
  geom_line() +
  geom_line(data = centres, mapping = aes(group = 1), colour = "black", alpha = 0.5) +
  facet_wrap(~ Cluster)

figCount <- incCount(figCount, "f_mfuzzClusters")
```

To recover common expression patterns in the meristem samples, read counts were transformed using the variance-stabilizing transformation (VST) included in the `DESeq2` package [@Love:2014do], and the standardised, VST-transformed read counts were clustered using the fuzzy *c*-means algorithm implemented in the `Mfuzz` package [@Futschik:2005vo]. To determine the number of cluster cores (*c*), the minimum distance between cluster centroids, the formation of empty clusters, and PCA plots of cluster members were monitored for clusters produced with *c* values between `cmin` and `cmax` (**SI**). A cluster number of `r as.character(english::english(c))` *justify chosing this? double diffential was closest to zero?* (**`r I(pasteLabel("Figure", figCount, "f_mfuzzClusters"))`**).

To understand the contents of the clusters, **z-score for each tf family by cluster**.

<a id="f_mfuzzClusters"></a>
```{r f_mfuzzClusters, fig.height=4.40933}
print(f_mfuzzClusters)
```

+ **Figure**: Gene expression clusters
+ GO enrichment? NO! z-score enrichment of transcription factor genes per cluster. Use dhyper in R (see http://jura.wi.mit.edu/bio/education/hot_topics/enrichment/Gene_list_enrichment_Mar10.pdf or http://stats.stackexchange.com/a/16259)
+ Clusters including 'known' genes

## Patterns of expression of transcription families

+ **Figure**: GSEA of transcription factor families

Because of the enrichment of TF family members in some of the gene expression clusters, geneset-level enrichment analysis (GSEA) was used to compare the levels of expression of TF family members in the meristem samples. **Something like: Consistent with their enrichment in cluster 1 and 6, MADS genes were significantly enriched in the SM/FM samples**. TF families involved in branching were enriched in **which families, which samples**.

## MADS genes: comparison of expression & homology in rice, tomato, arabidopsis

+ **Figure**: Similarity tree of MADS proteins with expression values and dN/dS values for each one.

# Discussion

# Experimental procedures

# Accession Numbers

# Acknowledgements

# Short legends for Supporting Information

# References

***

# Tables

***

# Figure legends

`r I(pasteLabel("Figure", figCount, "f_mfuzzClusters"))`
**Fuzzy *c*-means clustering of normalised, transformed read counts**

`r capwords(as.character(english::english(c)))` common patterns of expression were recovered from normalised, variance-stabilised read counts of expressed genes. Each line describes the expression pattern of one gene, with the gene's membership to the cluster mapped to the colour of the line. The core values for each cluster are drawn in black. 

***

# Figures

***

# Junk

I think `inline code` should work?