---
title: Genome-wide changes in expression during inflorescence architecture development
  in rice
csl: csl/the-plant-journal.csl
output:
  pdf_document:
    fig_caption: yes
    keep_md: yes
    keep_tex: yes
    number_sections: yes
  word_document:
    fig_caption: yes
  BiocStyle::html_document:
    fig_caption: yes
    force_captions: yes
    highlight: pygments
    keep_md: yes
    number_sections: yes
bibliography: bib/test.bib
---

Authors:

Addresses:

### Running title

### Keywords

### SRA details for libaries

Word count

***

# Summary

# Significance statement

# Introduction

* Rice inflorescence development
+ branching & grain yield
+ sequential changes in meristem activity
+ few key genes

# Results

## Dissections in early stages of inflorescence development

+  **Figure**: laser microdissections

## Transcriptome sequencing of meristem samples

+ Max. 1 paragraph about boring sequencing stats
+ **Table** (or supplementary): # reads, mapping %, rRNA, introns, genes blah blah

## Clustering of expression patterns

To recover common expression patterns in the meristem samples, read counts were transformed using the variance-stabilizing transformation (VST) included in the `DESeq2` package [@Love:2014do], and the standardised, VST-transformed read counts were clustered using the fuzzy *c*-means algorithm implemented in the `Mfuzz` package [@Futschik:2005vo]. To determine the number of cluster cores (*c*), the minimum distance between cluster centroids, the formation of empty clusters, and PCA plots of cluster members were monitored for clusters produced with *c* values between `cmin` and `cmax` (@SI). A cluster number of `{r} c` was chosen, 

+ **Figure**: Gene expression clusters
+ GO enrichment?
+ Clusters including 'known' genes

## Patterns of expression of transcription families
+ **Figure**: GSEA of transcription factor families

## MADS genes: comparison of expression & homology in rice, tomato, arabidopsis
+ **Figure**: Similarity tree of MADS proteins with expression values and dN/dS values for each one.

# Discussion

# Experimental procedures

# Accession Numbers

# Acknowledgements

# Short legends for Supporting Information

# References

***

# Tables

***

# Figure legends

***

# Figures

***

# Junk

I think `inline code` should work?