---
title: Genome-wide changes in expression during inflorescence architecture development
  in rice
csl: csl/the-plant-journal.csl
header-includes: \usepackage{graphicx} \usepackage[utf8]{inputenc} \DeclareUnicodeCharacter{2265}{$\geq$}
output:
  BiocStyle::html_document:
    dev: CairoPDF
    fig_caption: yes
    force_captions: yes
    highlight: pygments
    number_sections: yes
    toc: yes
  word_document:
    fig_caption: yes
    reference_docx: csl/wordStyles.docx
  pdf_document:
    fig_caption: yes
    number_sections: yes
  rmarkdown::tufte_handout: default
bibliography: bib/papersLib.bibtex
---

```{r Setup, include=FALSE, results="hide", warning=FALSE}

knitr::opts_chunk$set(fig.path='figures/',
               cache.path='cache/',
               dev=c("png", "cairo_pdf"),
               fig.width=3.150,
               fig.height=8.661,
               fig.align = 'center',
               dpi=300,
               fig.show='hold',
               cache=TRUE,
               echo=FALSE,
               results="hide",
               message=FALSE,
               warning=FALSE)

pander::panderOptions("table.caption.prefix", "")

source('src/R/figures.R')

```

Thomas W.R. Harrop^§^[^1], Israr Ud Din^¶^[^1], Veronica Gregis^¶^, Michela Osnato^¶^, Stefan Jouannic^§^, Hélène Adam^§^ and Martin Kater^¶^

[^1]: These authors contributed equally to this work.

| ^§^ Institut de Recherche pour le Développement, 911 Avenue Agropolis, 34394 Montpellier, France.
| ^¶^ Università degli Studi di Milano, Via Celoria 26, 20133 Milano, Italy.

## Running title

## Keywords

## SRA details for libraries

## Word count

# Summary

# Significance statement

# Introduction

## Rice inflorescence development
## Branching & grain yield
## Sequential changes in meristem activity
## Discuss a few key genes?

# Results and Discussion

To investigate gene expression during inflorescence development, LASER microdissection (LMD) was used to collect meristematic tissues from the rachis meristem (RM), primary branch meristem (PBM), elongating primary branch meristem and secondary branch meristem (ePBM/SBM) and spikelet meristem (SM) (`r I(pasteLabel("Figure ", figCount, "f_lmdFigure"))`). RNA isolated from the meristem samples was amplified and used to produce cDNA libraries for sequencing. **Need area cut +/- number of panicles used for each library from Israr**. RNA input for library preparation was between `r min(st_libStats$"Yield (ng)")` and `r max(st_libStats$"Yield (ng)")` ng and the average RNA integrity number was `r round(mean(st_libStats$"RIN"), 1)` (`r I(pasteLabel("Table S", s_tableCount, "st_libStats"))`). Initially, two biological replicates were prepared for each stage, but after performing a principal components analysis (PCA) on transformed expression values, two libraries that were produced from degraded RNA were excluded from the analysis (`r I(pasteLabel("Figure S", s_figCount, "sf_pca"))`). Following this, further samples were prepared to provide a total of three biological replicates with intact RNA (RIN ≥ 7) for each meristem type.

At least `r round(min(st_libStats$"Reads (M)"), 0)` million single-end, 50-base reads were produced from each library, yielding between `r round(min(st_libStats$"Reads in genes (M)", na.rm = TRUE), 0)` and `r round(max(st_libStats$"Reads in genes (M)", na.rm = TRUE), 0)` million reads that mapped to genes after excluding reads likely to have originated from rRNA and tRNA. Using strict cut-offs for gene expression, `r length(unique(readRDS("output/expressedGenes/expressedGenesAll.Rds")))` unique genes were detected in at least one of the libraries. As a preliminary assessment of the results, the qualitative and quantitative expression of a selection of genes were compared with patterns that had previously been reported (`r I(pasteLabel("Table S", s_tableCount, "st_reviewInSitu"))` and `r I(pasteLabel("Figure S", s_figCount, "sf_isGenesTpm"))`), and four genes that were detected in specific meristem samples were selected for validation by *in situ* hybridisation (**Figure Sx**). [**How do in situ / published results compare to transcriptome?**]

##  Patterns of gene expression during inflorescence development

To recover common expression patterns in expressed genes, read counts were transformed using the variance-stabilizing transformation (VST) included in the `DESeq2` package [@Love:2014do]. Expressed genes were ranked by variance, and standardised, VST-transformed read counts for the 25% most-variable genes were clustered using the fuzzy *c*-means algorithm implemented in the `Mfuzz` package [@Kumar:2007uw]. To determine the number of cluster cores (*c*), the minimum distance between cluster centroids, the formation of empty clusters, and PCA plots of cluster members were monitored for clusters produced with *c* values between 2 and `r maxClust` (`r I(pasteLabel("Figures S", s_figCount, "sf_mfuzzCentroids"))` `r I(pasteLabel(" and S", s_figCount, "sf_mfuzzPca"))`), leading to the use of `r as.character(english::english(c))` cluster cores (`r I(pasteLabel("Figure ", figCount, "f_mfuzzClusters"))`). Several common patterns were recovered by clustering: genes that increase or decrease in expression steadily during development (clusters `r getClusterName('inc', 'inc')` and `r getClusterName('dec', 'dec')`), genes that change expression between RM and PBM and remain constant from PBM to SM (clusters `r getClusterName('inc', 'small')` and `r getClusterName('dec', 'small')`), genes that have constant expression between RM and PBM and then increase or decrease in expression between PBM and SM (clusters `r getClusterName('small', 'inc')` and `r getClusterName('small', 'dec')`). There were also two weaker clusters containing fewer genes, which had complex expression patterns involving changes in expression particular to the PBM (clusters `r getClusterName('inc', 'dec')` and `r getClusterName('dec', 'inc')`).

Clusters `r getClusterName('dec', 'small')` and `r getClusterName('inc', 'small')` contain genes that change in expression between apical (RM) and axillary meristems (PBM–SM). Cluster `r getClusterName('dec', 'small')` contains three uncharacterized genes with annotations relating to auxin (*IAA10*, an auxin-responsive transcriptional regulator; *ABCB27*, an ABC transporter; and *LOC_Os01g63770*, an AUX/LAX transporter), and the GH3 gene *GH3-8*. [**Role of auxin influx / efflux in axillary / apical meristem specification**]. In *A. thaliana*, excess auxin may be inactivated by conjugation to amino acids by GH3 enzymes [@Staswick:2005cv]. Transgenic plants containing 3 kb of upstream sequence from *FLO-LFY HOMOLOG OF RICE* (*RFL*) fused to the *GH3-8* coding sequence, presumably expressing *GH3-8* in the endogenous pattern of *RFL* (*i.e.* PBM and SBM; [@IkedaKawakatsu:2012co]), have smaller panicles with fewer branches [@Yadav:2011db]. GH3-8 enzyme conjugates amino acids to auxin *in vitro* [@Ding:2008hv], suggesting that this phenotype may be correlated to reduced auxin signalling. [**Conclusion about auxin signalling**.]

**Auxin-related genes in clusters 2 & 4:**

+ *GH3-8*, *IAA10*, *ABCB27*, *LOC_Os01g63770* (AUX/LAX transporter) in cluster 2
+ *ARF6A*, *SAUR33*, *MDR12* (ABC), *ALS1* (ABC) in cluster 4

Two homologs of *LONELY GUY 1* (*LOG1*), *LONELY GUY LIKE PHOSPHORIBOHYDROLASE 1* (*LOGL1*) and *LONG AND BARBED AWN1* (*LABA1*/*LOGL6*/*An-2*), are present in cluster `r getClusterName('inc', 'inc')` with genes that are increasingly expressed between RM and SM. The LOG1 enzyme converts cytokinin nucleotides to the active form *in vitro*, and *log1* mutants produce small panicles with small SAMs and branching defects, and have lower expression of two cytokinin-inducible *RESPONSE REGULATOR* genes [@Kurakawa:2007go]. Reduced expression of a cytokinin oxidase gene, *GRAIN NUMBER 1A* (*GN1A*/*CKX2*), which encodes an enzyme that inactivates cytokinin, is associated with increased grain production in several high-yield rice varieties [@Ashikari:2005eg]. These results suggest that an increase in active cytokinin is associated with grain yield via an effect on panicle branching, but the function of *LABA1* suggests the opposite. The LABA1 protein present in *O. rufipogon*, which has an identical primary sequence to the protein from *O. sativa* ssp. *japonica* cv. Nipponbare, also activates cytokinin *in vitro* [@Gu:2015hy; @Hua:2015bs], and introduction of the *Laba1* allele from *O. rufipogon* into an *O. sativa* ssp *indica* background that contains a non-functional *laba1* allele results in a higher concentration of endogenous cytokinins and higher expression of *RESPONSE REGULATOR* genes [@Hua:2015bs], but reduces the number of grains per panicle and the number of tillers per plant [@Gu:2015hy]. *laba1* is a domestication allele found at high frequencies in cultivated accessions, implying that it was affected by artificial selection either for awn phenotype or grain yield [@Gu:2015hy; @Hua:2015bs]. No reports were found of *LOGL1* function, or the function of the gene annotated as a cytokinin dehydrogenase (*LOC_Os05g31040*) that is present in the complementary cluster (cluster `r getClusterName('dec', 'dec')`), but the expression pattern of *LABA1* and these two genes suggests increasing cytokinin activation and decreasing inactivation from RM to SM. It is not currently clear how such a mechanism might influence meristem identity or panicle branching at the molecular level. [**Check for at/tomato papers**]. There are also genes that may be involved in cytokinin signalling in other clusters, including a type-A *RESPONSE REGULATOR* (*RR3*) and a cytokinin receptor (*PHP2*) in cluster `r getClusterName('dec', 'small')`, containing genes more highly expressed in RM, suggesting that the role of cytokinins in inflorescence development may be complex.

Examination of the clustered genes suggested that transcription factors were overrepresented in most clusters (**annotated cluster tables in SI**). To determine whether particular TF families were associated with particular patterns of expression, the enrichment of TF family members in each cluster was calculated using hypergeometric tests. Using the list of transcription factor and other transcriptional regulator families available on the Plant Transcription Factor Database [@PerezRodriguez:2010co], a hypergeometric test was carried out if there were at least two members of one family in a given cluster. After adjustment for multiple comparisons, each cluster had a significant enrichment of at least one family (`r I(pasteLabel("Table ", tableCount, "t_hypergeom"))`). Cluster `r getClusterName('inc', 'small')`, containing genes that are more highly expressed in the PBM–SM samples than in the RM sample, has an enrichment of Homeobox and basic helix-loop-helix TFs whereas the cluster with the complementary pattern (cluster `r getClusterName('dec', 'small')`) is enriched for ABI3VP1 and G2-like TFs. MADS TFs and TRAF genes were overrepresented among genes that have low expression in the SM and PBM but higher expression in the RM (cluster `r getClusterName('small', 'inc')`) and bHLH, SBP and C2C2-GATA TFs were overreprented among genes with the inverse pattern (cluster `r getClusterName('small', 'dec')`)

The presence and co-expression of genes that are involved in inflorescence development along with other uncharacterized TF genes indicates that the clusters capture biologically meaningful groups of genes. Only three G1-Like (G1L/ALOG) genes were detected in the dataset. Two of them, *G1L2* and *G1L5* (*TAWAWA1*), were present in clusters `r getClusterName('dec', 'small')` and `r getClusterName('dec', 'dec')` respectively, and the third, *G1L1*, was not clustered but appears to have a similar expression pattern (`r I(pasteLabel("Figure ", figCount, "f_alogFamily"))`). Overexpression of *G1L5* increases the production of secondary branches and leads to the formation of tertiary branches, potentially by suppressing the transition from BM to SM identity, resulting in higher grain yield [@Yoshida:2013ff]. *G1L5* missense mutants have small inflorescences with reduced grain yield, but this phenotype was stronger when *G1L5* was targeted by RNAi [@Yoshida:2013ff]. This might be explained by redundancy between the three co-expressed G1L genes, *G1L1*, *G1L2* and *G1L5*, with all three of these genes being affected by the RNAi construct used against *G1L5*. [**Downstream targets of G1L5 (MADS22/MADS47/MADS55)**]

The AP2 gene *SUPERNUMERARY BRACT* (*SNB*) is involved in the transition between SM and floret meristem (FM) that occurs later in panicle development [@Lee:2007cj], but is coexpressed with *G1L5* in cluster `r getClusterName('dec', 'dec')` suggesting that it may also be involved in earlier transitions. Two other AP2 genes, *INDETERMINATE SPIKELET*, which acts redundantly with *SNB* to promote transition from SM to FM [@Lee:2012hj], and *FRIZZY PANICLE*, which is required for determination of SM [@Komatsu:2003ko], were found in clusters containing genes that are highly expressed in the SM.

| **I picked out some genes from the clusters (list directly below), are there any others in here I should discuss?**

1. Cluster 1
  + FZP
  + WRKY81
  + OsFBX127, OsFBX291
  + MADS5, MADS56
  + BZIP46, BZIP62
  + YAB6
  + LOC_Os07g02800 (MYB family?), LOC_Os07g12910 (PHD protein?)
2. Cluster 2
  + G1L2
  + SPL19
  + MADS47
  + HOX13
  + OsFBX30, OsFBO13, OsFBX258
3. Cluster 3
  + SNB
  + G1L5 (TAWAWA)
  + EP3 (OsFBK5), OsFBX257, OsFBX263, OsFBX333, OsFBX334, OsFBX421
  + MADS57
  + BZIP79
  + ARF9, SAUR25, OsSAUR48
  + WRKY36
  + BC10
  + LOC_Os12g39640 (MYB family?)
4. Cluster 4
  + PLT9 (AP2-like)
  + MADS61
  + ARF6A, OsSAUR33, LOC_Os08g44750 (auxin-induced protein 5NG4)
  + OsFBX18
  + BZIP23
  + ROC1, ROC3, LOC_Os09g35760 (homeobox/START)
5. Cluster 5
  + IDS1, ERF61 (AP2)
  + OsFBX109
  + LOC_Os08g25799 (MYB family?)
6. Cluster 6
  + SPL3, SPL12
  + WRKY123
  + MIR319A (leaf morphology / cold tolerance in indica: 10.1111/pce.12130. Nothing about inflorescence. Strange results, check.)
  + OsFBX14, OsFBL4, OsFBLD1
  + ARF2
  + C3H22
  + LOC_Os04g35430 (PHD domain?)
7. Cluster 7
  + LOGL1, LOGL6
  + BZIP47
  + HOX14
  + WRKY88
  + SHR1
  + SCR (GRAS)
  + NAC41 (No apical meristem protein)
  + OsFBO2, OsFBX125, OsFBX349
  + LOC_Os03g53700 (PHD domain?)
  + LOC_Os08g06240 (MYB domain?)
8. Cluster 8
  + PAP2 (MADS34)
  + DEP1
  + OsFBK1
  + SAUR6
  + LOC_Os02g13310 (homeobox?)

## Dynamic expression of transcription factor families

Because of the enrichment of TF family members in some of the gene expression clusters, geneset enrichment analysis (GSEA) was used to provide an overview of expression of transcription factors and other regulators in the meristem samples (`r I(pasteLabel("Figure ", figCount, "f_gsea"))`). Unlike soft clustering, which groups genes based on shared expression dynamics, this analysis was used to investigate the overall strength of expression of the families based on Log~2~-fold changes (L~2~FC) in read count of their members. There are some notable modes of expression shared between different families that may indicate roles in inflorescence development. The most pronounced change in expression appears to occur between the RM and branch meristem samples. Several families (including TUBBY-like, GROWTH-REGULATING FACTOR, bZIP and ABI3VP1 TF genes and Aux/IAA genes) are strongly expressed in RM and lower in the other meristem types. Another group, including mTERF, MYB, SBP and FHA TF genes and SET, PHD and Jumonji regulator genes, is enriched in one or both of the PBM and ePBM/SBM samples but depleted or not enriched in the RM samples.

The strong expression of SQUAMOSA promoter binding protein-like (SPL/SBP) genes in the RM and PBM and their presence in clusters `r getClusterName('small', 'dec')` and `r getClusterName('dec', 'small')` suggests involvement in the early stages of inflorescence development. Overexpression of *WEALTHY FARMER'S PANICLE* (*WFP*/*SPL14* or *IDEAL PLANT ARCHITECTURE 1*/*IPA1*), which was detected by *in situ* hybridisation experiments in the RM and branch meristems in wild-type plants, is associated with increased branching and grain yield [@Jiao:2010ft; @Miura:2010it]. **Downstream targets of WFP**. *WFP*, *SPL3*, *SPL12* and *SPL17* are all highly expressed in the RM, gradually decrease in expression and have weakest expression in the SM. These genes are all putative targets of *miR156*, a microRNA that causes reduced panicle size and delayed flowering when overexpressed [@Xie:2006dc].

These patterns are not intended to represent each member of each gene family. For example, there are several MADS genes that have lower expression in the SM than in the RM, but the test statistic for the whole MADS family was higher in the SM than in the RM. An example of this is the MADS gene *PANICLE PHYTOMER2* (*PAP2*/*MADS34*), which was strongly expressed at all stages with higher expression in the RM and PBM. *PAP2* acts with another *AGAMOUS-LIKE2*-like (*AGL2*-like/*SEPELLATA*) MADS gene, *LEAFY HULL STERILE 1* (*LHS1*/*MADS1*), to specify floral organs [@Gao:2010iz], but *pap2* mutants also have altered primary and secondary branch number, suggesting an earlier role in spikelet meristem specification [@Gao:2010iz; @Kobayashi:2010hn]. In contrast, *LHS1*, which is involved in floral meristem determination and floral organ development [@Jeon:2000tj; @Agrawal:2005ho], was detected only in the SM samples and is thus representative the overall enrichment of other MADS genes in those samples.

## Comparative analysis of MADS gene expression in eudicots and monocots

The enrichment of MADS genes in the SM sample and their overrepresentation in cluster `r getClusterName('small', 'inc')` corresponds with their involvement in flower development in rice and other plants. In the eudicot *A. thaliana*, *APETALA1* (*AP1*) is involved in determining FM identity, and members of the *AGL2*-like clade interact with other MADS genes to specify floral organ identity, such as *APETALA3* (*AP3*) and *PISTILLATA* (*PI*), which combine with *APETALA2* (*AP2*) to specify petals and another MADS gene, *AGAMOUS* (*AG*), to specify stamen (recently reviewed by @Prunet:2013ki). In rice, functions homologous to those provided by *A. thaliana* *PI* and *AP3* genes, related to specification of stamens and lodicules, appear to be performed redundantly by *MADS2*, *MADS4* and *SUPERWOMAN 1* (*SPW1*/*MADS16*) [**based on? check ref**], whilst stamen and carpels are specified by the expression of *RICE AGAMOUS* (*RAG*/*MADS3*) and *MADS58* [**check, add ref**]. [**Add info about tomato MADS**].

* Something about MADS genes involved earlier in development?
* Expression of MADS genes that interact with *G1L5*? (Maybe this is too much detail?)

A comparative analysis of expression of MIKC MADS genes in undifferentiated and differentiated meristems was used to investigate the evolutionary relationships between subclades in eudicots and monocots, exploiting two recent LMD RNA-sequencing datasets in addition to the one described in this paper. The RM and SM samples from rice inflorescences were compared with sequencing reads generated from the inflorescence meristem (IM) and FM of *A. thaliana* and the sympodial inflorescence meristem (SIM) and FM of the domesticated tomato, *Solanum lycopersicum* [@Park:2012dg; @Mantegazza:2014di]. A family- and clade-level overview of potential divergences, such as neo- and sub-functionalisation, was produced by combining a neighbour-joining similarity tree of MIKC MADS proteins with the change of expression of genes encoding them between indeterminate and determinate meristem tissues (`r I(pasteLabel("Figure ", figCount, "f_madsTree"))`). [**Note about relative genome size (number of genes) and MADS family size between species**.]

As expected based on their functions in floral organ specification in rice and *A. thaliana*, genes in the *AGL2*-like clade are generally more highly expressed in determinate meristems. However, a pair of tomato paralogs, apparently homologous to *SEP1* and *SEP2*, do not follow this pattern, which could indicate a role earlier or later in development or an unrelated function [**check SOLYC05G012020.2**].

* *AGL6*-like: AGL6 has a defined function but it doesn't look like other genes in the clade
* *STMADS11*-like:
* *SQUA*-like:
* *DEF*-like:
* *GLO*-like:
* *FLC*-like:

# Experimental procedures

## Plant material and sampling

*Oryza sativa* ssp. *japonica* cv. Nipponbare plants were grown in a growth chamber with 70% relative humidity at 30°C during the day and 26°C at night. After 7 weeks in long day conditions (14 hours light and 10 hours darkness) plants were transferred to short day conditions (10 hours light and 14 hours darkness) to induce floral transition. Inflorescence meristems were harvested from 6 to 14 days after the change of photoperiod.

## Tissue embedding, LASER microdissection and sequencing

Each inflorescence was harvested in 2 mL ice-cold 3:1 ethanol:acetic acid fixative solution, infiltrated (**filtered?**) twice under a mild vacuum for 15 minutes and stored for 20 hours in fresh fixative at 4°C. Embedding for dissection was performed as described by @Mantegazza:2014di. Tissues were cut into 8 µm sections on an RM2155 or RM2255 microtome (Leica) and dissected on a LMD6000 or LMD7000 LASER dissector (Leica). RNA was amplified using the ARCTURUS PicoPure RNA Isolation Kit (ThermoFisher) and assayed on a 2100 Bioanalyzer with the RNA 6000 Pico Kit (Agilent) before amplification and cDNA synthesis with the Ovation RNA-Seq System V2 (NuGEN). Library preparation with **which kit** and 50-base, single-end sequencing on the HiSeq 2000 platform (Illumina) were performed by IGA Technology Services (Udine, Italy).

## Data analyses

Analysis of the sequenced libraries was carried out with `bash` and `R` scripts [@RALanguageandEn:2015wf], which were arranged into a pipeline using the `ruffus` package for `python3` [@Goodstadt:2010jp]. All of the code used for the analysis and to generate text, figures and tables for this report, along with the versions and paramaters of the software used, are available with revision history in a public GitHub repository at https://github.com/evoreprice/lmdPaper.

Briefly, adaptor trimming was performed with `cutadapt` [@Martin:2011eu] and reads were mapped against the MSUv7 genome and annotation [@Kawahara:2013gh; @Ouyang:2007il] downloaded from Phytozome 10.3 [@Goodstein:2012ho] using `STAR` [@Dobin:2012fg] in 2-pass mode. The `'quantMode'` argument of `STAR` was used for read counting. rRNA and tRNA contamination was estimated using `htseq-count` [@Anders:2014gf] to count the number of reads that mapped to regions annotated as tRNA or rRNA on the Rap-DB [@Ohyanagi:2006go] or to regions where reads simulated with `wgsim` (https://github.com/lh3/wgsim) from rRNA sequences in the TIGR Plant Repeats Database [@Ouyang:2004ia] also mapped. Differential expression analysis to calculate Log~2~-fold change (L~2~FC) values and transformed read counts was performed with the `DESeq2` package [@Love:2014do]. To determine a strict cut-off for unexpressed genes, expression values in transcripts per million were calculated for each gene [@Li:2010cl; @Wagner:2012jf]. These values were compared to pseudo-expression values calculated from intergenic regions of the genome, using intervals of a similar size distribution to those used for calculating the 'genic' expression values. For each library, the cut-off was placed at the 95^th^ percentile of the distribution of intergenic expression values. Transformed counts and L~2~FC values from expressed genes were used for soft clustering with the `Mfuzz` package [@Kumar:2007uw] and geneset enrichment analysis with the `gage` package [@Luo:2009kp] respectively. All plots were produced with the `ggplot2` package [@Wickham:2009tn].

### Comparative analysis of MADS family

The primary sequences of *Arabidopsis thaliana*, *Solanum lycopersicum* and *Oryza sativa* MADS proteins were downloaded by querying Phytozome 10.3 [@Goodstein:2012ho] for the PANTHER family PTHR11945 ('MADS BOX PROTEIN') via the `biomaRt` interface [@Thomas:2003db; @Durinck:2009ki]. An initial alignment of all proteins was performed with `Clustal Omega` using a full distance matrix [@Sievers:2011fn]. To clean the alignment, poorly conserved regions where greater than `r minpcnongap` percent of sequences had gaps or less than `r minpcident` percent of pairwise comparisons between sequences had identical residues were removed. To recover MADS genes related to the Arabidopsis MIKC clade, a UGPMA tree was estimated from the cleaned alignment and the tree was cut at the lowest height that returned a clade including all type II Arabidopsis MADS proteins listed in the Arabidopsis transcription factor database on the Arabidopsis Gene Regulatory Information Server [@Yilmaz:2011ew], with as few type I proteins as possible. Protein sequences in this clade that were longer than `r minProtLength` amino acids were realigned using `Clustal Omega`, and the alignment was cleaned as described above and used to estimate a neighbour-joining tree with the `bionj` algorithm implemented in the `ape` package [@Paradis:2004dv] rooted to type I MADS sequences (`r og`). L~2~FC change expression values for tomato and Arabidopsis genes were estimated from samples that have previously been described [@Mantegazza:2014di; @Park:2012dg] using the same method described above for the rice samples. The tree was plotted using the `ggtree` extension for `ggplot2` (**ref ggtree**).

# Accession Numbers

# Acknowledgements

# Short legends for Supporting Information

**`r I(pasteLabel("Table S", s_tableCount, "st_libStats"))`**: Library and sequencing statistics.

**`r I(pasteLabel("Table S", s_tableCount, "st_reviewInSitu"))`**: Comparison of the dataset with published *in situ* expression profiles.

**`r I(pasteLabel("Figure S", s_figCount, "sf_pca"))`**: Principal components analysis of transformed read counts for each library.

**`r I(pasteLabel("Figure S", s_figCount, "sf_isGenesTpm"))`**: Quantitative expression values for the genes that were compared to published *in situ* results.

**`r I(pasteLabel("Figure S", s_figCount, "sf_mfuzzCentroids"))`**: Minimum centroid distance for 2–`r maxClust` cluster cores (*c*).

**`r I(pasteLabel("Figure S", s_figCount, "sf_mfuzzPca"))`**: Classical multidimensional scaling (MDS) of Euclidean distances between transformed read counts for clustered genes.

# Tables

```{r t_familiesPerCluster, results = "hide"}
pander::set.caption(paste0(
  I(pasteLabel("**Table ", tableCount, "t_familiesPerCluster")),
  "**: Transcription factor and transcriptional regulator family members in each cluster. *n*, total number of family members, *n*~expr~, number of family members expressed in the whole dataset, C~1~–C~", c, "~, number of family members in each cluster, and the total number of family members in all clusters."))
t_familiesPerCluster <- as.data.frame(t_familiesPerCluster[!Total == 0])
rownames(t_familiesPerCluster) <- t_familiesPerCluster$Family
t_familiesPerCluster$Family <- NULL
pander::pandoc.table(t_familiesPerCluster, split.tables = Inf, emphasize.rownames = FALSE,
                     justify = c("right", rep("centre", length(names(t_familiesPerCluster)) - 1), "left"))
```


```{r t_hypergeom, results = "asis"}
pander::set.alignment('center', row.names = 'right')
pander::set.caption(paste0(
  I(pasteLabel("**Table ", tableCount, "t_hypergeom")),
  "**: Transcription factor and transcriptional regulator family enrichment for families that had at least two members in any one cluster, including the number of family members (*n*), number expressed in the whole dataset (*n*~expr~) and number and adjusted *p*-value for each cluster."))
pander::pandoc.table(t_hypergeom, split.tables = Inf, emphasize.rownames = FALSE)
```

# Figure legends

----

![](data/lmdFigure.pdf)

**`r I(pasteLabel("Figure ", figCount, "f_lmdFigure"))`**. LASER microdissection of meristem samples in the early stages of inflorescence development. Samples were collected from the rachis meristem (**a** and **e**), the primary branch meristem (**b** and **f**), the elongating primary branch meristem and secondary branch meristem (**c** and **g**) and the spikelet meristem (**d** and **h**). Images show the samples before (**a**–**d**) and after (**e**–**h**) dissection.

----

```{r f_mfuzzClusters, fig.height = 8.661 / 2}
print(f_mfuzzClusters)
```

**`r I(pasteLabel("Figure ", figCount, "f_mfuzzClusters"))`**. Common patterns of gene expression in developing inflorescences. Fuzzy *c*-means clustering of normalised, variance-stabilised read counts was used to recover `r as.character(english::english(c))` common patterns of expression. Each line describes the expression pattern of one gene, with the gene's membership to the cluster represented by the colour of the line. The core values for each cluster are plotted in black. The ordering of the panels in the plot was rearranged to enable side-by-side comparison of complementary clusters.

----

```{r f_alogFamily, fig.width = 3.150, fig.height = 3.150 * 2}
print(f_alogFamily)
```

**`r I(pasteLabel("Figure ", figCount, "f_alogFamily"))`**. Expression of rice G1/ALOG family genes in inflorescence meristems. Genes were retrieved by querying Phytozome 10.3 for the Pfam domain PF04852 [@Goodstein:2012ho; @Finn:2014kj]. Red or blue points indicate whether the gene was detected or not detected respectively for each library. Only three G1 genes (*G1L1*, *G1L2* and *G1L5*/*TAWAWA*) were detected, but they share a similar pattern of expression.

----

```{r f_gsea, fig.height = 8.661 / 2}
print(f_gsea)
```

**`r I(pasteLabel("Figure ", figCount, "f_gsea"))`**. Geneset enrichment analysis (GSEA) for transcription factor and other transcriptional regulator families. Test statistics were calculated from the L~2~FC values between the stage of interest and all other stages for the expressed genes in each geneset. The absolute value of the test statistic indicates the magnitude of the geneset-level change, and a positive test statistic indicates enrichment and a negative statistic indicates depletion. Although some test statistics had low *p*-values none were lower than 0.10 after Benjamini-Hochberg correction.

----

```{r f_madsTree}
print(f_madsTree)
```

**`r I(pasteLabel("Figure ", figCount, "f_madsTree"))`**. Neighbour-joining similarity tree of putatative MIKC proteins in *A. thaliana*, *Solanum lycopersicum* and *Oryza sativa* ssp. *japonica*. Named proteins starting with AT_ or OS_ are from Arabidopsis or rice respectively, with synonyms in parentheses, and other proteins are labelled with their unique identifiers. The colour of the label represents the Log~2~-fold change (L~2~FC) in expression of the gene encoding that protein between undifferentiated and differentiated meristems. Where there is no coloured label, the gene for the protein in question was removed from the differential expression analysis by `DESeq2`, which may occur because of low read counts, an extreme count outlier or the gene failing independent filtering [@Love:2014do]. For rice, L~2~FC was estimated from the RM and SM samples reported here. For *A. thaliana*, L~2~FC was estimated using the inflorescence meristem and floret meristem samples reported by @Mantegazza:2014di. The sympodial inflorescence meristem and floret meristem samples generated by @Park:2012dg were used to estimate L~2~FC values for tomato genes. Clades are labelled following the nomenclature presented by [@Arora:2007fa].

----

# `sessionInfo` (*hide*)

```{r sessionInfo, results = "asis"}
cat(paste0("**branch:** _", system("git rev-parse --abbrev-ref HEAD", intern = TRUE), "_\n\n"))
cat(paste0("**hash:** _", system("git rev-parse HEAD", intern = TRUE), "_\n\n"))
pander::pander(sessionInfo())
```

# References (move to before **Tables**)
