---
title: "Supporting Information"
csl: csl/the-plant-journal.csl
header-includes: \usepackage{graphicx}
output:
  BiocStyle::html_document:
    fig_caption: yes
    force_captions: yes
    highlight: pygments
    keep_md: yes
    number_sections: yes
    dev: "CairoPDF"
  word_document:
    fig_caption: yes
  pdf_document:
    fig_caption: yes
    keep_tex: yes
    number_sections: yes
bibliography: bib/papersLib.bib
---

```{r Setup, include=FALSE, results="hide", warning=FALSE}

knitr::opts_chunk$set(fig.path='figures/',
               cache.path='cache/',
               dev=c("png", "cairo_pdf"),
               fig.align = 'center',
               dpi=300,
               fig.show='hold',
               cache=FALSE,
               echo=FALSE,
               results="hide",
               message=FALSE,
               warning=FALSE)

source('src/R/figures.R')

```

``` {r st_libStats, results = "asis"}
pander::set.alignment('left', row.names = 'right')
pander::set.caption(paste0(
  I(pasteLabel("**Table S", s_tableCount, "st_libStats")),
  "**: Library and sequencing statistics."))
pander::pandoc.table(st_libStats, split.tables = Inf, emphasize.rownames = FALSE, missing = "")
```

```{r st_reviewInSitu, results = "asis"}
pander::set.alignment('left', row.names = 'right')
pander::set.caption(paste0(
  I(pasteLabel("**Table S", s_tableCount, "st_reviewInSitu")),
  "**: Comparison of the dataset with published *in situ* expression profiles. For each tissue, a '1' indicates that the gene was detected by both methods, a '2' indicates that the gene was detected by *in situ* hybridisation but not by RNA sequencing and a '3' indicates that the gene was detected by RNA sequencing but not reported as expressed by *in situ*. The last category is not informative because *in situ* analysis for the stage in question may have been omitted from the publication. A blank cell indicates that the gene was not detected by either method."))
pander::pandoc.table(st_reviewInSitu, split.tables = Inf, emphasize.rownames = FALSE,
                     emphasize.cols = c(1,2), missing = "")
```

```{r sf_pca}
print(sf_pca)
```

**`r I(pasteLabel("Figure S", s_figCount, "sf_pca"))`**: Principal components analysis of transformed read counts for each library. The two libraries made from degraded RNA did not cluster with other libraries (see `r I(pasteLabel("Table S", s_tableCount, "st_libStats"))`), suggesting that the use of degraded RNA had an effect on the quantification of read counts. For this reason these two libraries were not included in subsequent analyses.

```{r sf_mfuzzCentroids}
print(sf_mfuzzCentroids)
```

**`r I(pasteLabel("Figure S", s_figCount, "sf_mfuzzCentroids"))`**: Minimum centroid distance for 2â€“`r maxClust` cluster cores (*c*). Eight cluster cores were used because the rate of change of the minimum centroid distance over *c* decreases after $c = 8$.

``` {r sf_mfuzzPca}
print(sf_mfuzzPca)
```

**`r I(pasteLabel("Figure S", s_figCount, "sf_mfuzzPca"))`**: Classical multidimensional scaling (MDS) of Euclidean distances between transformed read counts for clustered genes. Genes are coloured by the cluster to which they were assigned. The separation of genes by cluster on the MDS plot supports the use of eight clusters.

# References for supporting information

labels <- c("1" = "Detected\nin both", "2" = "Only detected\nby in situ", "3" = "Only detected by\nRNA-sequencing")
