---
title: ""
csl: csl/the-plant-journal.csl
date: "\\vspace{0 mm}"
header-includes:
  \usepackage{graphicx}
  \usepackage{caption}
  \usepackage{fixltx2e}
  \usepackage{rotating}
  \usepackage{siunitx}
  \renewcommand{\thefigure}{S\arabic{figure}}
  \renewcommand{\thetable}{S\arabic{table}}
documentclass: tufte-handout
classoption: a4paper
output:
  rmarkdown::tufte_handout:
    dev: CairoPDF
    keep_tex: yes
    fig_crop: no
bibliography: bib/papersLib.bibtex
---

```{r Setup, include=FALSE, results="hide", warning=FALSE}
knitr::opts_chunk$set(fig.path='figures/',
               cache.path='cache/',
               fig.align = 'center', 
               dpi=600,
               fig.show='hold',
               fig.height = 4.213,
               fig.width = 4.213,
               echo=FALSE,
               results="hide",
               message=FALSE,
               warning=FALSE)

source('src/R/figures.R')
```

```{r sf_pca, fig.cap = paste0("Principal components analysis of transformed read counts for each library. The two libraries made from degraded RNA did not cluster with other libraries (see Table \\ref{tab:st_libstats}), suggesting that the use of degraded RNA affected quantification of expression. For this reason these two libraries were not included in subsequent analyses.")}
print(sf_pca)
```

\clearpage
\captionof{table}{\protect\label{tab:st_libstats}Library and sequencing statistics.}

``` {r st_libstats, results = "asis"}
st_libStats_x <- as.data.frame(st_libStats)
rownames(st_libStats_x) <- st_libStats_x$Library
st_libStats_x$Library <- NULL
st_libStats_x$Sample <- sub("ePBM/SBM", "AM¹", st_libStats_x$Sample, fixed = TRUE)
st_libStats_x <- xtable::xtable(st_libStats_x)
xtable::align(st_libStats_x) <- rep("r", ncol(st_libStats_x) + 1)
xtable::digits(st_libStats_x) <- c(rep(0, 4), 3, rep(1, 11), 0)
st_libStats_x_print <- print(st_libStats_x,
      caption.placement = "top",
      comment = FALSE,
      rotate.colnames = TRUE,
      floating = TRUE,
      floating.environment = "table*",
      print.results = FALSE)

# remove the first hline
st_libStats_x_print <- sub("\\begin{tabular}{rrrrrrrrrrrrrrrr}\n  \\hline\n", "\\begin{tabular}{rrrrrrrrrrrrrrrr}\n", st_libStats_x_print, fixed = TRUE)
cat(st_libStats_x_print)
cat("¹ ePBM/AM")
```


```{r sf_isGenesTpm, fig.height = 9.535, fig.cap = paste0("\\protect\\label{fig:sf_isGenesTpm}","Comparison of the LMD RNA-sequencing dataset with published \\emph{in situ} results. Expression value in transcripts per million (TPM) is shown on the \\emph{y}-axis, and points are plotted as a triangle if the expression value was above the cutoff for that library or a circle if below. Grey shading indicates a meristem type for which expression was detected by \\emph{in situ} hybridisation. See Table \\ref{tab:refFrame_x} for references for the \\emph{in situ} expression patterns. Although four replicates were produced from each of the RM and PBM, only three replicates for each stage are included in the analysis (see Results).", "\\newline ", "Gene expression measured by RNAseq may differ from the qualitative patterns measured by RNA \\emph{in situ} hybridization for several reasons, including the relative sensitivity of the techniques, the specificity of the \\emph{in situ} probe, and differences in growth conditions (\\emph{e.g. HD5}) or the accession under study (\\emph{e.g. GN1A}).")}
print(sf_isGenesTpm)
```

\clearpage
\captionof{table}{\protect\label{tab:refFrame_x}References for Figure \ref{fig:sf_isGenesTpm}.}

```{r insituRefs, results = "asis"}
pander::pandoc.table(st_refTable, emphasize.cols = c(1,2))
```

```{r sf_mfuzzPcaCentroids, fig.height = 6.741, fig.cap = paste0("Assessment of clustering results. \\textbf{a}. Minimum centroid distance for 2–", maxClust, " cluster cores (\\emph{c}). Eight cluster cores were indicated because the rate of change of the minimum centroid distance over \\emph{c} decreases after \\emph{c} = 8. \\textbf{b}. Classical multidimensional scaling (MDS) of Euclidean distances between transformed read counts for clustered genes. Genes are coloured by the cluster to which they were assigned. The separation of genes by cluster on the MDS plot supports the use of eight clusters.")}
grid::grid.draw(sf_mfuzzPcaCentroids)
```

```{r sf_madsTree, fig.height = 9.535, fig.cap = paste0("Differential expression of genes encoding putative MIKC proteins in \\emph{A. thaliana}, \\emph{Solanum lycopersicum} and \\emph{Oryza sativa} ssp. \\emph{japonica}. The tree is a neighbour-joining tree estimated from genetic distance of amino acid sequences (see Methods S1). Named proteins starting with AT\\_ or OS\\_ are from Arabidopsis or rice respectively, with synonyms in parentheses, and other proteins are labelled with their unique identifiers.", "\\newline ", "The colour of the label represents the Log\\textsubscript{2}-fold change (L\\textsubscript{2}FC) in expression of the gene encoding that protein between undifferentiated and differentiated meristems. Where there is no coloured label, the adjusted \\emph{p}-value associated with the difference in expression was greater than 0.1, or the gene for the protein in question was excluded from the differential expression analysis by \\texttt{DESeq2}, which may occur because of low read counts, an extreme count outlier or the gene failing independent filtering (Love \\emph{et al}., 2014).", "\\newline ", "For rice, L\\textsubscript{2}FC was estimated from the RM and SM samples reported here. For \\emph{A. thaliana}, L\\textsubscript{2}FC was estimated using the inflorescence meristem and floret meristem samples reported by Mantegazza \\emph{et al}. (2014). The sympodial inflorescence meristem and floret meristem samples generated by Park \\emph{et al}. (2012) were used to estimate L\\textsubscript{2}FC values for tomato genes.")}
print(sf_madsTree)
```

\clearpage

# Methods S1

## Comparative analysis of the MADS family

The primary sequences of *Arabidopsis thaliana*, *Solanum lycopersicum* and *Oryza sativa* MADS proteins were downloaded by querying Phytozome 10.3 [@Goodstein:2012ho] for the PANTHER family PTHR11945 ('MADS BOX PROTEIN') via the `biomaRt` interface [@Thomas:2003db; @Durinck:2009ki]. An initial alignment of all proteins was performed with `Clustal Omega` using a full distance matrix [@Sievers:2011fn]. To clean the alignment, poorly conserved regions where greater than `r minpcnongap` percent of sequences had gaps or less than `r minpcident` percent of pairwise comparisons between sequences had identical residues were removed. To recover MADS genes related to the Arabidopsis MIKC clade, a UGPMA tree was estimated from the cleaned alignment and the tree was cut at the lowest height that returned a clade including all type II Arabidopsis MADS proteins listed in the Arabidopsis transcription factor database on the Arabidopsis Gene Regulatory Information Server [@Yilmaz:2011ew], with as few type I proteins as possible. Protein sequences in this clade that were longer than `r minProtLength` amino acids were realigned using `Clustal Omega`, and the alignment was cleaned as described above and used to estimate a neighbour-joining tree with the `bionj` algorithm implemented in the `ape` package [@Paradis:2004dv] rooted to type I MADS sequences (`r og`). L~2~FC change expression values for tomato and Arabidopsis genes were estimated from samples that have previously been described [@Mantegazza:2014di; @Park:2012dg] using the same `STAR`–`DESeq2` pipeline described in the main text for the rice samples [@Dobin:2012fg; @Love:2014do]. The tree was plotted using the `ggtree` extension for `ggplot2` [https://github.com/GuangchuangYu/ggtree; @Wickham:2009tn], with clade labels following the nomenclature presented by [@Arora:2007fa].

\clearpage

# References for supporting information
